"use strict";(self.webpackChunkwebsite_2=self.webpackChunkwebsite_2||[]).push([[6692],{3905:function(e,t,a){a.d(t,{Zo:function(){return m},kt:function(){return d}});var n=a(7294);function r(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function i(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function o(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?i(Object(a),!0).forEach((function(t){r(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):i(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function l(e,t){if(null==e)return{};var a,n,r=function(e,t){if(null==e)return{};var a,n,r={},i=Object.keys(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||(r[a]=e[a]);return r}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(r[a]=e[a])}return r}var s=n.createContext({}),p=function(e){var t=n.useContext(s),a=t;return e&&(a="function"==typeof e?e(t):o(o({},t),e)),a},m=function(e){var t=p(e.components);return n.createElement(s.Provider,{value:t},e.children)},u={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},c=n.forwardRef((function(e,t){var a=e.components,r=e.mdxType,i=e.originalType,s=e.parentName,m=l(e,["components","mdxType","originalType","parentName"]),c=p(a),d=r,h=c["".concat(s,".").concat(d)]||c[d]||u[d]||i;return a?n.createElement(h,o(o({ref:t},m),{},{components:a})):n.createElement(h,o({ref:t},m))}));function d(e,t){var a=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var i=a.length,o=new Array(i);o[0]=c;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l.mdxType="string"==typeof e?e:r,o[1]=l;for(var p=2;p<i;p++)o[p]=a[p];return n.createElement.apply(null,o)}return n.createElement.apply(null,a)}c.displayName="MDXCreateElement"},4822:function(e,t,a){a.r(t),a.d(t,{frontMatter:function(){return o},contentTitle:function(){return l},metadata:function(){return s},toc:function(){return p},default:function(){return u}});var n=a(7462),r=a(3366),i=(a(7294),a(3905)),o={sidebar_position:1},l="Exome Germline Single Sample Overview",s={unversionedId:"Pipelines/Exome_Germline_Single_Sample_Pipeline/README",id:"Pipelines/Exome_Germline_Single_Sample_Pipeline/README",isDocsHomePage:!1,title:"Exome Germline Single Sample Overview",description:"| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |",source:"@site/docs/Pipelines/Exome_Germline_Single_Sample_Pipeline/README.md",sourceDirName:"Pipelines/Exome_Germline_Single_Sample_Pipeline",slug:"/Pipelines/Exome_Germline_Single_Sample_Pipeline/README",permalink:"/warp/docs/Pipelines/Exome_Germline_Single_Sample_Pipeline/README",editUrl:"https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Exome_Germline_Single_Sample_Pipeline/README.md",tags:[],version:"current",lastUpdatedBy:"Denis Loginov",lastUpdatedAt:1633546218,formattedLastUpdatedAt:"10/6/2021",sidebarPosition:1,frontMatter:{sidebar_position:1},sidebar:"docsSidebar",previous:{title:"CEMBA_v1.1.0 Publication Methods",permalink:"/warp/docs/Pipelines/CEMBA_MethylC_Seq_Pipeline/CEMBA.methods"},next:{title:"Exome Germline Single Sample v2.4.4 Methods",permalink:"/warp/docs/Pipelines/Exome_Germline_Single_Sample_Pipeline/exome.methods"}},p=[{value:"Set-up",id:"set-up",children:[{value:"Workflow Installation and Requirements",id:"workflow-installation-and-requirements",children:[]},{value:"Software Version Requirements",id:"software-version-requirements",children:[]},{value:"Input Requirements and Expectations",id:"input-requirements-and-expectations",children:[]}]},{value:"Workflow Tasks and Tools",id:"workflow-tasks-and-tools",children:[]},{value:"Workflow Outputs",id:"workflow-outputs",children:[{value:"Base quality scores",id:"base-quality-scores",children:[]}]},{value:"Important Notes",id:"important-notes",children:[]},{value:"Contact Us",id:"contact-us",children:[]},{value:"Licensing",id:"licensing",children:[]}],m={toc:p};function u(e){var t=e.components,a=(0,r.Z)(e,["components"]);return(0,i.kt)("wrapper",(0,n.Z)({},m,a,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"exome-germline-single-sample-overview"},"Exome Germline Single Sample Overview"),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:"center"},"Pipeline Version"),(0,i.kt)("th",{parentName:"tr",align:"center"},"Date Updated"),(0,i.kt)("th",{parentName:"tr",align:"center"},"Documentation Author"),(0,i.kt)("th",{parentName:"tr",align:"center"},"Questions or Feedback"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:"center"},(0,i.kt)("a",{parentName:"td",href:"https://github.com/broadinstitute/warp/releases"},"ExomeGermlineSingleSample_v2.4.3")),(0,i.kt)("td",{parentName:"tr",align:"center"},"December 07, 2020"),(0,i.kt)("td",{parentName:"tr",align:"center"},(0,i.kt)("a",{parentName:"td",href:"mailto:ekiernan@broadinstitute.org"},"Elizabeth Kiernan")),(0,i.kt)("td",{parentName:"tr",align:"center"},"Please file GitHub issues in dsde-pipelines or contact ",(0,i.kt)("a",{parentName:"td",href:"mailto:kdegatano@broadinstitute.org"},"Kylee Degatano"))))),(0,i.kt)("p",null,"The Exome Germline Single Sample pipeline implements data pre-processing and initial variant calling according to the GATK Best Practices for germline SNP and Indel discovery in human exome sequencing data."),(0,i.kt)("p",null,"For a broad overview of the pipeline processes, read the GATK Best Practices documentation for ",(0,i.kt)("a",{parentName:"p",href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035535912"},"data pre-processing")," and for ",(0,i.kt)("a",{parentName:"p",href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035535932"},"germline short variant discovery"),"."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Want to try the Exome Germline Single Sample pipeline?")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"You can test the pipeline in Terra! Go the ",(0,i.kt)("a",{parentName:"p",href:"https://app.terra.bio/#workspaces/warp-pipelines/Exome-Analysis-Pipeline"},"Exome-Analysis-Pipeline workspace")," which includes sample data and workflows for preprocessing and initial variant calling, sample map generation, and joint genotyping."))),(0,i.kt)("h2",{id:"set-up"},"Set-up"),(0,i.kt)("h3",{id:"workflow-installation-and-requirements"},"Workflow Installation and Requirements"),(0,i.kt)("p",null,"The Exome Germline Single Sample workflow is written in the Workflow Description Language WDL and can be downloaded by cloning the ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/tree/master"},"warp repository")," in GitHub. The workflow can be deployed using ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/cromwell"},"Cromwell"),", a GA4GH compliant, flexible workflow management system that supports multiple computing platforms. For the latest workflow version and release notes, please see the Exome Germline Single Sample ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/pipelines/broad/dna_seq/germline/single_sample/exome/ExomeGermlineSingleSample.changelog.md"},"changelog"),"."),(0,i.kt)("h3",{id:"software-version-requirements"},"Software Version Requirements"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://github.com/broadinstitute/gatk/releases/tag/4.1.8.0"},"GATK 4.1.8.0")),(0,i.kt)("li",{parentName:"ul"},"Picard 2.23.8"),(0,i.kt)("li",{parentName:"ul"},"Samtools 1.11"),(0,i.kt)("li",{parentName:"ul"},"Python 3.0"),(0,i.kt)("li",{parentName:"ul"},"Cromwell version support",(0,i.kt)("ul",{parentName:"li"},(0,i.kt)("li",{parentName:"ul"},"Successfully tested on v52"),(0,i.kt)("li",{parentName:"ul"},"Does not work on versions < v23 due to output syntax"))),(0,i.kt)("li",{parentName:"ul"},"Papi version support",(0,i.kt)("ul",{parentName:"li"},(0,i.kt)("li",{parentName:"ul"},"Successfully tested on Papi v2")))),(0,i.kt)("h3",{id:"input-requirements-and-expectations"},"Input Requirements and Expectations"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"Human exome sequencing data in unmapped BAM (uBAM) format"),(0,i.kt)("li",{parentName:"ul"},"One or more read groups, one per uBAM file, all belonging to a single sample (SM)"),(0,i.kt)("li",{parentName:"ul"},"Input uBAM files must additionally comply with the following requirements:",(0,i.kt)("ul",{parentName:"li"},(0,i.kt)("li",{parentName:"ul"},'Filenames all have the same suffix (we use ".unmapped.bam")'),(0,i.kt)("li",{parentName:"ul"},"Files must pass validation by ValidateSamFile"),(0,i.kt)("li",{parentName:"ul"},"Reads are provided in query-sorted order"),(0,i.kt)("li",{parentName:"ul"},"All reads must have an RG tag"))),(0,i.kt)("li",{parentName:"ul"},'GVCF output names must end in ".g.vcf.gz"'),(0,i.kt)("li",{parentName:"ul"},"Reference genome must be Hg38 with ALT contigs"),(0,i.kt)("li",{parentName:"ul"},"Unique exome calling, target, and bait ",(0,i.kt)("a",{parentName:"li",href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035531852"},".interval_list")," obtained from the sequencing provider. Generally the calling, target, and bait files will not be the same.")),(0,i.kt)("h2",{id:"workflow-tasks-and-tools"},"Workflow Tasks and Tools"),(0,i.kt)("p",null,"The Exome Germline Single Sample ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/pipelines/broad/dna_seq/germline/single_sample/exome/ExomeGermlineSingleSample.wdl"},"workflow")," imports a series of tasks from the WARP ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/tree/master/tasks/broad"},"tasks library")," and a DNASeq struct (",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/structs/dna_seq/DNASeqStructs.wdl"},"DNASeqStructs.wdl"),") containing reference files from the ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/tree/master/structs/dna_seq"},"structs library"),"."),(0,i.kt)("p",null,"You can read more about the software tools implemented in these tasks by reading the GATK ",(0,i.kt)("a",{parentName:"p",href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035535912"},"data pre-processing")," and ",(0,i.kt)("a",{parentName:"p",href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035535932"},"germline short variant discovery")," documentation."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Want to use the Exome Germline Single Sample workflow in your publication?")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Check out the workflow ",(0,i.kt)("a",{parentName:"p",href:"/warp/docs/Pipelines/Exome_Germline_Single_Sample_Pipeline/exome.methods"},"Methods")," to get started!"))),(0,i.kt)("h2",{id:"workflow-outputs"},"Workflow Outputs"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"CRAM, CRAM index, and CRAM MD5"),(0,i.kt)("li",{parentName:"ul"},"GVCF and its GVCF index"),(0,i.kt)("li",{parentName:"ul"},"BQSR report"),(0,i.kt)("li",{parentName:"ul"},"Summary metrics; to read more about any particular metric, you can search the metric using the ",(0,i.kt)("a",{parentName:"li",href:"https://gatk.broadinstitute.org/hc/en-us/categories/360002302312"},"GATK documentation search"))),(0,i.kt)("h3",{id:"base-quality-scores"},"Base quality scores"),(0,i.kt)("p",null,"The final CRAM files have base quality scores binned according to the ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/CCDG/Pipeline-Standardization/blob/master/PipelineStandard.md#base-quality-score-binning-scheme"},"Functional Equivalence specification")," (",(0,i.kt)("a",{parentName:"p",href:"https://www.nature.com/articles/s41467-018-06159-4"},"Regier et al., 2018"),")."),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:null},"Original Score"),(0,i.kt)("th",{parentName:"tr",align:null},"Score after BQSR recalibration"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"1-6"),(0,i.kt)("td",{parentName:"tr",align:null},"unchanged")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"7-12"),(0,i.kt)("td",{parentName:"tr",align:null},"10")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"13-22"),(0,i.kt)("td",{parentName:"tr",align:null},"20")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"22-infinity"),(0,i.kt)("td",{parentName:"tr",align:null},"30")))),(0,i.kt)("h2",{id:"important-notes"},"Important Notes"),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"Runtime parameters are optimized for Broad's Google Cloud Platform implementation."),(0,i.kt)("li",{parentName:"ul"},"For help running workflows on the Google Cloud Platform or locally please\nview the following tutorial ",(0,i.kt)("a",{parentName:"li",href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035530952"},"(How to) Execute Workflows from the gatk-workflows Git Organization"),"."),(0,i.kt)("li",{parentName:"ul"},"Please visit the ",(0,i.kt)("a",{parentName:"li",href:"https://gatk.broadinstitute.org/hc/en-us/categories/360002310591"},"GATK Technical Documentation")," site for further documentation on our workflows and tools."),(0,i.kt)("li",{parentName:"ul"},"You can access relevant reference and resource bundles in the ",(0,i.kt)("a",{parentName:"li",href:"https://gatk.broadinstitute.org/hc/en-us/articles/360035890811"},"GATK Resource Bundle"),".")),(0,i.kt)("h2",{id:"contact-us"},"Contact Us"),(0,i.kt)("p",null,"This material is provided by the Data Science Platform group at the Broad Institute. Please direct any questions or concerns to one of our forum sites : ",(0,i.kt)("a",{parentName:"p",href:"https://gatk.broadinstitute.org/hc/en-us/community/topics"},"GATK")," or ",(0,i.kt)("a",{parentName:"p",href:"https://support.terra.bio/hc/en-us/community/topics/360000500432"},"Terra"),"."),(0,i.kt)("h2",{id:"licensing"},"Licensing"),(0,i.kt)("p",null,"Copyright Broad Institute, 2020 | BSD-3"),(0,i.kt)("p",null,"The workflow script is released under the ",(0,i.kt)("strong",{parentName:"p"},"WDL open source code license (BSD-3)")," (full license text at ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/LICENSE"},"https://github.com/broadinstitute/warp/blob/master/LICENSE"),"). However, please note that the programs it calls may be subject to different licenses. Users are responsible for checking that they are authorized to run all programs before running this script."),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://software.broadinstitute.org/gatk/download/licensing.php"},"GATK")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"http://bio-bwa.sourceforge.net/bwa.shtml#13"},"BWA")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://broadinstitute.github.io/picard/"},"Picard")),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"http://www.htslib.org/terms/"},"Samtools"))))}u.isMDXComponent=!0}}]);