"use strict";(self.webpackChunkwebsite_2=self.webpackChunkwebsite_2||[]).push([[8601],{3905:function(e,t,a){a.d(t,{Zo:function(){return m},kt:function(){return c}});var n=a(7294);function r(e,t,a){return t in e?Object.defineProperty(e,t,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[t]=a,e}function i(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),a.push.apply(a,n)}return a}function l(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?i(Object(a),!0).forEach((function(t){r(e,t,a[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):i(Object(a)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))}))}return e}function p(e,t){if(null==e)return{};var a,n,r=function(e,t){if(null==e)return{};var a,n,r={},i=Object.keys(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||(r[a]=e[a]);return r}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)a=i[n],t.indexOf(a)>=0||Object.prototype.propertyIsEnumerable.call(e,a)&&(r[a]=e[a])}return r}var o=n.createContext({}),s=function(e){var t=n.useContext(o),a=t;return e&&(a="function"==typeof e?e(t):l(l({},t),e)),a},m=function(e){var t=s(e.components);return n.createElement(o.Provider,{value:t},e.children)},d={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},u=n.forwardRef((function(e,t){var a=e.components,r=e.mdxType,i=e.originalType,o=e.parentName,m=p(e,["components","mdxType","originalType","parentName"]),u=s(a),c=r,h=u["".concat(o,".").concat(c)]||u[c]||d[c]||i;return a?n.createElement(h,l(l({ref:t},m),{},{components:a})):n.createElement(h,l({ref:t},m))}));function c(e,t){var a=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var i=a.length,l=new Array(i);l[0]=u;var p={};for(var o in t)hasOwnProperty.call(t,o)&&(p[o]=t[o]);p.originalType=e,p.mdxType="string"==typeof e?e:r,l[1]=p;for(var s=2;s<i;s++)l[s]=a[s];return n.createElement.apply(null,l)}return n.createElement.apply(null,a)}u.displayName="MDXCreateElement"},848:function(e,t,a){a.r(t),a.d(t,{frontMatter:function(){return l},contentTitle:function(){return p},metadata:function(){return o},toc:function(){return s},default:function(){return d}});var n=a(7462),r=a(3366),i=(a(7294),a(3905)),l={sidebar_position:1},p="Smart-seq2 Multi-Sample Overview",o={unversionedId:"Pipelines/Smart-seq2_Multi_Sample_Pipeline/README",id:"Pipelines/Smart-seq2_Multi_Sample_Pipeline/README",isDocsHomePage:!1,title:"Smart-seq2 Multi-Sample Overview",description:"| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |",source:"@site/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README.md",sourceDirName:"Pipelines/Smart-seq2_Multi_Sample_Pipeline",slug:"/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README",permalink:"/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README",editUrl:"https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/README.md",tags:[],version:"current",lastUpdatedBy:"kew24",lastUpdatedAt:1668631540,formattedLastUpdatedAt:"11/16/2022",sidebarPosition:1,frontMatter:{sidebar_position:1},sidebar:"docsSidebar",previous:{title:"scATAC v1.2.0 Methods",permalink:"/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/scatac.methods"},next:{title:"Smart-seq2 Multi-Sample v2.2.1 Publication Methods",permalink:"/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/smart-seq2.methods"}},s=[{value:"Introduction",id:"introduction",children:[]},{value:"Inputs",id:"inputs",children:[{value:"Sample and Reference Inputs",id:"sample-and-reference-inputs",children:[]},{value:"Additional Input",id:"additional-input",children:[]},{value:"Smart-seq2 Multi-Sample Task Summary",id:"smart-seq2-multi-sample-task-summary",children:[]},{value:"Outputs",id:"outputs",children:[]}]},{value:"Validation",id:"validation",children:[]},{value:"Versioning",id:"versioning",children:[]},{value:"Citing the Smart-seq2 Multi-Sample Pipeline",id:"citing-the-smart-seq2-multi-sample-pipeline",children:[]},{value:"Consortia Support",id:"consortia-support",children:[]},{value:"Have Suggestions?",id:"have-suggestions",children:[]}],m={toc:s};function d(e){var t=e.components,a=(0,r.Z)(e,["components"]);return(0,i.kt)("wrapper",(0,n.Z)({},m,a,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("h1",{id:"smart-seq2-multi-sample-overview"},"Smart-seq2 Multi-Sample Overview"),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:"center"},"Pipeline Version"),(0,i.kt)("th",{parentName:"tr",align:"center"},"Date Updated"),(0,i.kt)("th",{parentName:"tr",align:"center"},"Documentation Author"),(0,i.kt)("th",{parentName:"tr",align:"center"},"Questions or Feedback"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:"center"},(0,i.kt)("a",{parentName:"td",href:"https://github.com/broadinstitute/warp/releases"},"MultiSampleSmartSeq2_v2.2.1")),(0,i.kt)("td",{parentName:"tr",align:"center"},"May, 2021"),(0,i.kt)("td",{parentName:"tr",align:"center"},(0,i.kt)("a",{parentName:"td",href:"mailto:ekiernan@broadinstitute.org"},"Elizabeth Kiernan")),(0,i.kt)("td",{parentName:"tr",align:"center"},"Please file GitHub issues in WARP or contact ",(0,i.kt)("a",{parentName:"td",href:"mailto:warp-pipelines-help@broadinstitute.org"},"the WARP team"))))),(0,i.kt)("h2",{id:"introduction"},"Introduction"),(0,i.kt)("p",null,"The Smart-seq2 Multi-Sample (Multi-SS2) Pipeline is a wrapper around the ",(0,i.kt)("a",{parentName:"p",href:"../Smart-seq2_Single_Sample_Pipeline/README"},"Smart-seq2 Single Sample")," pipeline. It is developed by the Data Coordination Platform of the Human Cell Atlas to process single-cell RNAseq (scRNAseq) data generated by Smart-seq2 assays. The workflow processes multiple cells by importing and running the ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl"},"Smart-seq2 Single Sample workflow")," for each cell (sample) and then merging the resulting Loom matrix output into a single Loom matrix containing raw counts and TPMs."),(0,i.kt)("p",null,"Full details about the Smart-seq2 Pipeline can be read in the ",(0,i.kt)("a",{parentName:"p",href:"../Smart-seq2_Single_Sample_Pipeline/README"},"Smart-seq2 Single Sample Overview")," in GitHub."),(0,i.kt)("p",null,"The Multi-SS2 workflow can also be run in ",(0,i.kt)("a",{parentName:"p",href:"https://app.terra.bio"},"Terra"),", a cloud-based analysis platform. The Terra ",(0,i.kt)("a",{parentName:"p",href:"https://app.terra.bio/#workspaces/featured-workspaces-hca/HCA%20Smart-seq2%20Multi%20Sample%20Pipeline"},"Smart-seq2 public workspace")," contains the Smart-seq2 workflow, workflow configurations, required reference data and other inputs, and example testing data."),(0,i.kt)("div",{className:"admonition admonition-tip alert alert--success"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"}))),"Want to use the Multi-SS2 Pipeline for your publication?")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Check out the ",(0,i.kt)("a",{parentName:"p",href:"/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/smart-seq2.methods"},"Smart-seq2 Publication Methods")," to get started!"))),(0,i.kt)("h2",{id:"inputs"},"Inputs"),(0,i.kt)("p",null,"There are two example configuration (JSON) files available for testing the Multi-SS2 workflow. Both examples are also preloaded in the Terra ",(0,i.kt)("a",{parentName:"p",href:"https://app.terra.bio/#workspaces/featured-workspaces-hca/HCA%20Smart-seq2%20Multi%20Sample%20Pipeline"},"Smart-seq2 public workspace"),"."),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_multisample/human_single_example.json"},"human_single_example.json"),": Configurations for an example single-end human dataset consisting of two samples (cells)"),(0,i.kt)("li",{parentName:"ul"},(0,i.kt)("a",{parentName:"li",href:"https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_multisample/mouse_paired_example.json"},"mouse_paired_example.json"),": Configurations for an example paired-end mouse dataset consisting of two samples (cells)")),(0,i.kt)("h3",{id:"sample-and-reference-inputs"},"Sample and Reference Inputs"),(0,i.kt)("p",null,"The workflow\u2019s sample inputs are listed in the table below. Reference inputs are identical to those\nspecified in the ",(0,i.kt)("a",{parentName:"p",href:"../Smart-seq2_Single_Sample_Pipeline/README"},"Smart-seq2 Single Sample Overview"),"."),(0,i.kt)("p",null,"The workflow processes both single- and paired-end samples; however, these samples can not be mixed in the same run."),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:null},"Input name"),(0,i.kt)("th",{parentName:"tr",align:null},"Input Description"),(0,i.kt)("th",{parentName:"tr",align:null},"Input Type"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"fastq1_input_files"),(0,i.kt)("td",{parentName:"tr",align:null},"Cloud locations for each read1 file"),(0,i.kt)("td",{parentName:"tr",align:null},"Array of strings")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"fastq2_input_files"),(0,i.kt)("td",{parentName:"tr",align:null},"Optional cloud locations for each read2 file if running paired-end samples"),(0,i.kt)("td",{parentName:"tr",align:null},"Array of strings")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"input_ids"),(0,i.kt)("td",{parentName:"tr",align:null},"Unique identifiers or names for each cell; can be a UUID or human-readable name"),(0,i.kt)("td",{parentName:"tr",align:null},"Array of strings")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"input_names"),(0,i.kt)("td",{parentName:"tr",align:null},"Optional unique identifiers/names to further describe each cell. If ",(0,i.kt)("inlineCode",{parentName:"td"},"input_id")," is a UUID, the ",(0,i.kt)("inlineCode",{parentName:"td"},"input_name")," could be used as a human-readable identifier"),(0,i.kt)("td",{parentName:"tr",align:null},"String")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"batch_id"),(0,i.kt)("td",{parentName:"tr",align:null},"Identifier for the batch of multiple samples"),(0,i.kt)("td",{parentName:"tr",align:null},"String")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"batch_name"),(0,i.kt)("td",{parentName:"tr",align:null},"Optional string to describe the batch or biological sample"),(0,i.kt)("td",{parentName:"tr",align:null},"String")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"input_name_metadata_field"),(0,i.kt)("td",{parentName:"tr",align:null},"Optional input describing, when applicable, the metadata field containing the ",(0,i.kt)("inlineCode",{parentName:"td"},"input_names")),(0,i.kt)("td",{parentName:"tr",align:null},"String")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"input_id_metadata_field"),(0,i.kt)("td",{parentName:"tr",align:null},"Optional string describing, when applicable, the metadata field containing the ",(0,i.kt)("inlineCode",{parentName:"td"},"input_ids")),(0,i.kt)("td",{parentName:"tr",align:null},"String")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"project_id")),(0,i.kt)("td",{parentName:"tr",align:null},"Optional project identifier; usually a number"),(0,i.kt)("td",{parentName:"tr",align:null},"String")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"project_name")),(0,i.kt)("td",{parentName:"tr",align:null},"Optional project identifier; usually a human-readable name"),(0,i.kt)("td",{parentName:"tr",align:null},"String")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"library")),(0,i.kt)("td",{parentName:"tr",align:null},"Optional description of the sequencing method or approach"),(0,i.kt)("td",{parentName:"tr",align:null},"String")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"organ")),(0,i.kt)("td",{parentName:"tr",align:null},"Optional description of the organ from which the cells were derived"),(0,i.kt)("td",{parentName:"tr",align:null},"String")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"species")),(0,i.kt)("td",{parentName:"tr",align:null},"Optional description of the species from which the cells were derived"),(0,i.kt)("td",{parentName:"tr",align:null},"String")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("inlineCode",{parentName:"td"},"paired-end")),(0,i.kt)("td",{parentName:"tr",align:null},"Boolean for whether samples are paired-end or not"),(0,i.kt)("td",{parentName:"tr",align:null},"Boolean")))),(0,i.kt)("h3",{id:"additional-input"},"Additional Input"),(0,i.kt)("p",null,'The reference inputs are identical to those specified in the "Additional Reference Inputs" section of the ',(0,i.kt)("a",{parentName:"p",href:"https://broadinstitute.github.io/warp/documentation/Pipelines/Smart-seq2_Single_Sample_Pipeline/#inputs"},"Smart-seq2 Single Sample Overview"),"."),(0,i.kt)("h3",{id:"smart-seq2-multi-sample-task-summary"},"Smart-seq2 Multi-Sample Task Summary"),(0,i.kt)("p",null,"The Multi-SS2 Pipeline calls three tasks:"),(0,i.kt)("p",null,"1) ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl"},"SmartSeq2SingleSample.wdl"),": a task that runs the Smart-seq2 Single Sample workflow\n2) ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/tasks/SmartSeq2PlateAggregation.wdl"},"SmartSeq2PlateAggregation.wdl"),": the wrapper pipeline that checks the inputs, initiates per cell processing and aggregates the results\n3) ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/tasks/LoomUtils.wdl"},"LoomUtils.wdl"),": a task that produces the Loom matrix output containing raw counts and TPMs"),(0,i.kt)("h3",{id:"outputs"},"Outputs"),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:null},"Output file name"),(0,i.kt)("th",{parentName:"tr",align:null},"Output Description"),(0,i.kt)("th",{parentName:"tr",align:null},"Output Type"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"bam_files"),(0,i.kt)("td",{parentName:"tr",align:null},"An array of genome-aligned BAM files (one for each sample) generated with HISAT2"),(0,i.kt)("td",{parentName:"tr",align:null},"Array")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"bam_index_files"),(0,i.kt)("td",{parentName:"tr",align:null},"An array of BAM index files generated with HISAT2"),(0,i.kt)("td",{parentName:"tr",align:null},"Array")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"loom_output"),(0,i.kt)("td",{parentName:"tr",align:null},"A single Loom cell-by-gene matrix containing raw counts and TPMs for every cell"),(0,i.kt)("td",{parentName:"tr",align:null},"File")))),(0,i.kt)("p",null,"The final Loom matrix is an aggregate of all the individual Loom matrices generated using the ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl"},"Smart-seq2 Single Sample workflow"),"."),(0,i.kt)("p",null,"The aggregated Loom filename contains the ",(0,i.kt)("inlineCode",{parentName:"p"},"batch_id")," prefix, which is the string specified in the input configuration. The ",(0,i.kt)("inlineCode",{parentName:"p"},"batch_id")," is also set as a global attribute in the Loom."),(0,i.kt)("p",null,"Both the individual sample Loom files and individual BAM files are described in the ",(0,i.kt)("a",{parentName:"p",href:"../Smart-seq2_Single_Sample_Pipeline/README"},"Smart-seq2 Single Sample Overview"),"."),(0,i.kt)("div",{className:"admonition admonition-warning alert alert--danger"},(0,i.kt)("div",{parentName:"div",className:"admonition-heading"},(0,i.kt)("h5",{parentName:"div"},(0,i.kt)("span",{parentName:"h5",className:"admonition-icon"},(0,i.kt)("svg",{parentName:"span",xmlns:"http://www.w3.org/2000/svg",width:"12",height:"16",viewBox:"0 0 12 16"},(0,i.kt)("path",{parentName:"svg",fillRule:"evenodd",d:"M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"}))),"Zarr Array Deprecation Notice June 2020")),(0,i.kt)("div",{parentName:"div",className:"admonition-content"},(0,i.kt)("p",{parentName:"div"},"Please note that we have deprecated the previously used Zarr array output. The pipeline now uses the Loom file format as the default output."))),(0,i.kt)("h2",{id:"validation"},"Validation"),(0,i.kt)("p",null,"The Multi-SS2 Pipeline has been validated for processing human and mouse, stranded or unstranded, paired- or single-end, and plate- or fluidigm-based Smart-seq2 datasets (see links to validation reports in the table below)."),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:null},"Workflow Configuration"),(0,i.kt)("th",{parentName:"tr",align:null},"Link to Report"))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Mouse paired-end"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("a",{parentName:"td",href:"https://docs.google.com/document/d/12zGTFROrcXEByt9z0h06qjSqb9vWutn28Tx6YiND1Ds/edit"},"Report"))),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Human and mouse single-end"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("a",{parentName:"td",href:"https://docs.google.com/document/d/1MonsTG8UnROHZ_XpulrSZNTxO988KEH6T6h45plFYQg/edit#heading=h.ixoqmhbabdvh"},"Report"))),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Human stranded fluidigm"),(0,i.kt)("td",{parentName:"tr",align:null},(0,i.kt)("a",{parentName:"td",href:"https://docs.google.com/document/d/1FEg86Tlu657j9Kjw_v3keFQRXcBIs8gOqCwLbPSP-C0/edit#heading=h.wjr8otl7zg14"},"Report"))))),(0,i.kt)("h2",{id:"versioning"},"Versioning"),(0,i.kt)("p",null,"Release information for the Multi-SS2 Pipeline can be found in the ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_multisample/MultiSampleSmartSeq2.changelog.md"},"changelog"),". Please note that any major changes to the Smart-seq2 pipeline will be documented in the ",(0,i.kt)("a",{parentName:"p",href:"https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.changelog.md"},"Smart-seq2 Single Sample changelog"),"."),(0,i.kt)("h2",{id:"citing-the-smart-seq2-multi-sample-pipeline"},"Citing the Smart-seq2 Multi-Sample Pipeline"),(0,i.kt)("p",null,"Please identify the pipeline in your methods section using the Smart-seq2 Multi-Sample Pipeline's ",(0,i.kt)("a",{parentName:"p",href:"https://scicrunch.org/scicrunch/Resources/record/nlx_144509-1/SCR_018920/resolver?q=Smart-seq2&l=Smart-seq2"},"SciCrunch resource identifier"),"."),(0,i.kt)("ul",null,(0,i.kt)("li",{parentName:"ul"},"Ex: ",(0,i.kt)("em",{parentName:"li"},"Smart-seq2 Multi-Sample Pipeline (RRID:SCR_018920)"))),(0,i.kt)("h2",{id:"consortia-support"},"Consortia Support"),(0,i.kt)("p",null,"This pipeline is supported and used by the ",(0,i.kt)("a",{parentName:"p",href:"https://www.humancellatlas.org/"},"Human Cell Atlas")," (HCA) project. "),(0,i.kt)("p",null,"If your organization also uses this pipeline, we would love to list you! Please reach out to us by contacting ",(0,i.kt)("a",{parentName:"p",href:"mailto:warp-pipelines-help@broadinstitute.org"},"the WARP team"),"."),(0,i.kt)("h2",{id:"have-suggestions"},"Have Suggestions?"),(0,i.kt)("p",null,"Please help us make our tools better by contacting ",(0,i.kt)("a",{parentName:"p",href:"mailto:warp-pipelines-help@broadinstitute.org"},"the WARP team")," for pipeline-related suggestions or questions."))}d.isMDXComponent=!0}}]);