<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.5">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">Smart-seq2 Single Nucleus Multi-Sample Overview | WARP</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/README"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Smart-seq2 Single Nucleus Multi-Sample Overview | WARP"><meta data-react-helmet="true" name="description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><meta data-react-helmet="true" property="og:description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/README"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/README" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/README" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.57c5b218.css">
<link rel="preload" href="/warp/assets/js/runtime~main.90c71525.js" as="script">
<link rel="preload" href="/warp/assets/js/main.1fa25a74.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_3Zt9 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">🌜</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">🌞</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button class="clean-btn backToTopButton_35hR" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed menuWithAnnouncementBar_2WvA"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">About WARP</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">WARP Pipelines</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">CEMBA</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Exome Germline Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">GDC Whole Genome Somatic Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Illumina Genotyping Array</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Imputation</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Optimus</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Single Cell ATAC</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Multi-Sample</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/README">Smart-seq2 Single Nucleus Multi-Sample Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/multi_snss2.methods">Smart-seq2 Single Nucleus Multi-Sample v1.1.0 Publication Methods</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Whole Genome Germline Single Sample</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Contribution Guide</a></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_1CGd"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_3E-R"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="tocCollapsible_1PrD tocMobile_3Hoh"><button type="button" class="clean-btn tocCollapsibleButton_2O1e">On this page</button></div><div class="markdown"><header><h1>Smart-seq2 Single Nucleus Multi-Sample Overview</h1></header><table><thead><tr><th align="center">Pipeline Version</th><th align="center">Date Updated</th><th align="center">Documentation Author</th><th align="center">Questions or Feedback</th></tr></thead><tbody><tr><td align="center"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">MultiSampleSmartSeq2SingleNuclei_v1.1.0.</a></td><td align="center">July, 2021</td><td align="center"><a href="mailto:ekiernan@broadinstitute.org" target="_blank" rel="noopener noreferrer">Elizabeth Kiernan</a></td><td align="center">Please file GitHub issues in WARP or contact <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a></td></tr></tbody></table><p><img src="/warp/assets/images/snSS2-49fa29b5c051bd785b5af73abd3bfd35.png"></p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="introduction-to-the-smart-seq2-single-nucleus-multi-sample-pipeline"></a>Introduction to the Smart-seq2 Single Nucleus Multi-Sample pipeline<a class="hash-link" href="#introduction-to-the-smart-seq2-single-nucleus-multi-sample-pipeline" title="Direct link to heading">#</a></h2><p>The Smart-seq2 Single Nucleus Multi-Sample (Multi-snSS2) pipeline was developed in collaboration with the <a href="https://biccn.org/" target="_blank" rel="noopener noreferrer">BRAIN Initiative Cell Census Network</a> (BICCN) to process single-nucleus RNAseq (snRNAseq) data generated by <a href="https://www.nature.com/articles/nmeth.2639" target="_blank" rel="noopener noreferrer">Smart-seq2 assays</a>. The pipeline&#x27;s workflow is written in WDL, is freely available in the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/smartseq2_single_nucleus_multisample/MultiSampleSmartSeq2SingleNucleus.wdl" target="_blank" rel="noopener noreferrer">WARP repository</a> on GitHub, and can be run by any compliant WDL runner (e.g. <a href="https://github.com/broadinstitute/cromwell" target="_blank" rel="noopener noreferrer">Crowmell</a>). </p><p>The pipeline is designed to process snRNA-seq data from multiple cells. Overall, the workflow trims paired-end FASTQ files, aligns reads to the genome using a modified GTF, counts intronic and exonic reads, and calculates quality control metrics.</p><p>The pipeline has been scientifically validated by the BRAIN Institute. Read more in the <a href="#validation">validation section</a>.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Try the Multi-snSS2 workflow in Terra</h5></div><div class="admonition-content"><p>You can run the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_nucleus_multisample/MultiSampleSmartSeq2SingleNucleus.wdl" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Nucleus Multi-Sample workflow</a> in <a href="https://app.terra.bio" target="_blank" rel="noopener noreferrer">Terra</a>, a cloud-based analysis platform. The Terra <a href="https://app.terra.bio/#workspaces/warp-pipelines/Smart-seq2_Single_Nucleus_Muti-Sample" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Nucleus Multi-Sample public workspace</a> is preloaded with the Multi-snSS2 workflow, example testing data, and all the required reference data.</p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="quick-start-table"></a>Quick start table<a class="hash-link" href="#quick-start-table" title="Direct link to heading">#</a></h2><table><thead><tr><th>Pipeline features</th><th>Description</th><th>Source</th></tr></thead><tbody><tr><td>Assay type</td><td>Smart-seq2 Single Nucleus</td><td><a href="https://www.nature.com/articles/nprot.2014.006" target="_blank" rel="noopener noreferrer">Smart-seq2</a></td></tr><tr><td>Overall workflow</td><td>Quality control and transcriptome quantification.</td><td>Code available from the <a href="https://github.com/broadinstitute/warp/tree/develop/pipelines/skylab/smartseq2_single_nucleus/SmartSeq2SingleNucleus.wdl" target="_blank" rel="noopener noreferrer">WARP repository</a> in GitHub</td></tr><tr><td>Workflow language</td><td>WDL</td><td><a href="https://github.com/openwdl/wdl" target="_blank" rel="noopener noreferrer">openWDL</a></td></tr><tr><td>Genomic reference sequence (for validation)</td><td>GRCm38 mouse genome primary sequence.</td><td>GENCODE GRCm38 <a href="http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M23/GRCm38.primary_assembly.genome.fa.gz" target="_blank" rel="noopener noreferrer">Mouse</a></td></tr><tr><td>Transcriptomic reference annotation (for validation)</td><td>Modified <a href="https://www.gencodegenes.org/mouse/release_M23.html" target="_blank" rel="noopener noreferrer">M23</a> GTF built with the  <a href="https://github.com/broadinstitute/warp/tree/master/pipelines/skylab/build_indices/BuildIndices.wdl" target="_blank" rel="noopener noreferrer">BuildIndices workflow</a>.</td><td>GENCODE</td></tr><tr><td>Aligner</td><td>STAR (v.2.7.9a)</td><td><a href="https://github.com/alexdobin/STAR" target="_blank" rel="noopener noreferrer">STAR</a></td></tr><tr><td>QC metrics</td><td>Picard (v.2.20.4)</td><td><a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Broad Institute</a></td></tr><tr><td>Transcript quantification</td><td>featureCounts (utilities for counting reads to genomic features).</td><td><a href="http://subread.sourceforge.net/" target="_blank" rel="noopener noreferrer">featureCounts(v2.0.2)</a></td></tr><tr><td>Data input file format</td><td>File format in which sequencing data is provided.</td><td><a href="https://academic.oup.com/nar/article/38/6/1767/3112533" target="_blank" rel="noopener noreferrer">FASTQ</a></td></tr><tr><td>Data output file formats</td><td>File formats in which Smart-seq2 output is provided.</td><td><a href="http://samtools.github.io/hts-specs/" target="_blank" rel="noopener noreferrer">BAM</a>, Loom (counts and metrics; generated with <a href="http://loompy.org/" target="_blank" rel="noopener noreferrer">Loompy v.3.0.6)</a>, TSV (counts)</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="set-up"></a>Set-Up<a class="hash-link" href="#set-up" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="multi-snss2-installation-and-requirements"></a>Multi-snSS2 installation and requirements<a class="hash-link" href="#multi-snss2-installation-and-requirements" title="Direct link to heading">#</a></h3><p>The latest release of the workflow, example data, and dependencies are available from the WARP <a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">releases page</a> (see release tags prefixed with SmartSeq2SingleNucleus). To discover and search releases, use the WARP command-line tool <a href="https://github.com/broadinstitute/warp/tree/develop/wreleaser" target="_blank" rel="noopener noreferrer">Wreleaser</a>.
. </p><p>The workflow is deployed using <a href="https://github.com/broadinstitute/cromwell" target="_blank" rel="noopener noreferrer">Cromwell</a>, a GA4GH compliant, flexible workflow management system that supports multiple computing platforms.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="inputs"></a>Inputs<a class="hash-link" href="#inputs" title="Direct link to heading">#</a></h3><p>There is an <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/smartseq2_single_nucleus_multisample/mouse_example.json" target="_blank" rel="noopener noreferrer">example configuration (JSON) file</a> available that you can to test the Multi-snSS2 workflow. It points to publicly available reference files and sample paired-end FASTQs.</p><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="sample-data-and-reference-inputs"></a>Sample data and reference inputs<a class="hash-link" href="#sample-data-and-reference-inputs" title="Direct link to heading">#</a></h4><p>The table below details the Multi-snSS2 inputs. The pipeline is designed to take in an array of paired-end reads in the form of two FASTQ files per cell. </p><ul><li>Reference inputs are created using the <a href="https://github.com/broadinstitute/warp/tree/master/pipelines/skylab/build_indices" target="_blank" rel="noopener noreferrer">BuildIndices Pipeline</a>.</li><li>The workflow uses a modified version of the 10x Genomic&#x27;s code for building mouse (<a href="https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#mm10_2020A" target="_blank" rel="noopener noreferrer">GRCm38-2020-A</a>) and human (<a href="https://support.10xgenomics.com/single-cell-gene-expression/software/release-notes/build#GRCh38_2020A" target="_blank" rel="noopener noreferrer">GRCh38-2020-A</a>) reference packages. </li><li>To enable intron counting, the workflow calls a <a href="https://github.com/broadinstitute/warp/blob/develop/dockers/skylab/snss2-build-indices/add-introns-to-gtf.py" target="_blank" rel="noopener noreferrer">shell script</a> to create a custom GTF with intron annotations. Introns are considered any part of a contig that is not exonic nor intergenic. </li></ul><table><thead><tr><th>Input Name</th><th>Input Description</th><th>Input Format</th></tr></thead><tbody><tr><td>fastq1_input_files</td><td>Cloud path to FASTQ files containing forward paired-end sequencing reads for each cell (sample); order must match the order in input_id.</td><td>Array of strings</td></tr><tr><td>fastq2_input_files</td><td>Cloud path to FASTQ files containing reverse paired-end sequencing reads for each cell (sample); order must match the order in input_id.</td><td>Array of strings</td></tr><tr><td>input_ids</td><td>Unique identifiers or names for each cell; can be a UUID or human-readable name.</td><td>Array of strings</td></tr><tr><td>input_names</td><td>Optional unique identifiers/names to further describe each cell. If <code>input_ids</code> is a UUID, the <code>input_names</code> could be used as a human-readable identifier.</td><td>String</td></tr><tr><td>batch_id</td><td>Identifier for the batch of multiple samples.</td><td>String</td></tr><tr><td>batch_name</td><td>Optional string to describe the batch or biological sample.</td><td>String</td></tr><tr><td>input_name_metadata_field</td><td>Optional input describing, when applicable, the metadata field containing the <code>input_names</code>.</td><td>String</td></tr><tr><td>input_id_metadata_field</td><td>Optional string describing, when applicable, the metadata field containing the <code>input_ids</code>.</td><td>String</td></tr><tr><td>project_id</td><td>Optional project identifier; usually a number.</td><td>String</td></tr><tr><td>project_name</td><td>Optional project identifier; usually a human-readable name.</td><td>String</td></tr><tr><td>library</td><td>Optional description of the sequencing method or approach.</td><td>String</td></tr><tr><td>organ</td><td>Optional description of the organ from which the cells were derived.</td><td>String</td></tr><tr><td>species</td><td>Optional description of the species from which the cells were derived.</td><td>String</td></tr><tr><td>tar_star_reference</td><td>Genome references for STAR alignment.</td><td>TAR</td></tr><tr><td>annotations_gtf</td><td>Custom GTF file containing annotations for exon and intron tagging; must match the STAR reference.</td><td>GTF</td></tr><tr><td>genome_ref_fasta</td><td>FASTA file used for STAR alignment.</td><td>FASTA</td></tr><tr><td>adapter_list</td><td>File listing adapter sequences used in the library preparation (i.e. Illumina adapters for Illumina sequencing).</td><td>FASTA</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="running-multi-snss2"></a>Running Multi-snSS2<a class="hash-link" href="#running-multi-snss2" title="Direct link to heading">#</a></h2><p>The <a href="https://github.com/broadinstitute/warp/tree/develop/pipelines/skylab/smartseq2_single_nucleus/SmartSeq2SingleNucleus.wdl" target="_blank" rel="noopener noreferrer">Multi-snSS2 workflow </a> is in the <a href="https://github.com/broadinstitute/warp/tree/snSS2_first_wdls/pipelines/skylab/smartseq2_single_nucleus" target="_blank" rel="noopener noreferrer">pipelines/smartseq2_single_nucleus folder</a> of the WARP repository and implements the workflow by importing individual tasks (written in WDL script) from the WARP <a href="https://github.com/broadinstitute/warp/tree/master/tasks/skylab/" target="_blank" rel="noopener noreferrer">tasks folder</a>.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="multi-snss2-workflow-summary"></a>Multi-snSS2 workflow summary<a class="hash-link" href="#multi-snss2-workflow-summary" title="Direct link to heading">#</a></h3><table><thead><tr><th>Task name and task’s WDL link</th><th>Description</th><th>Software</th><th>Tool</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/skylab/CheckInputs.wdl" target="_blank" rel="noopener noreferrer">CheckInputs.checkInputArrays</a></td><td>Checks the inputs and initiates the per cell processing.</td><td>Bash</td><td>NA</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/TrimAdapters.wdl" target="_blank" rel="noopener noreferrer">TrimAdapters.TrimAdapters</a></td><td>Trims adapter sequences from the FASTQ inputs</td><td><a href="https://github.com/ExpressionAnalysis/ea-utils" target="_blank" rel="noopener noreferrer">ea-utils</a>.</td><td><a href="https://github.com/ExpressionAnalysis/ea-utils/tree/master/clipper" target="_blank" rel="noopener noreferrer">fastq-mcf</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/StarAlign.wdl" target="_blank" rel="noopener noreferrer">StarAlign.StarAlignFastqMultisample</a></td><td>Aligns reads to the genome.</td><td>STAR</td><td>STAR</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/Picard.wdl" target="_blank" rel="noopener noreferrer">Picard.RemoveDuplicatesFromBam</a></td><td>Removes duplicate reads, producing a new BAM output; adds regroups to deduplicated BAM.</td><td>Picard</td><td>MarkDuplicates, AddOrReplaceReadGroups</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/Picard.wdl" target="_blank" rel="noopener noreferrer">Picard.CollectMultipleMetricsMultiSample</a></td><td>Collects QC metrics on the deduplicated BAM files.</td><td>Picard</td><td>CollectMultipleMetrics</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/FeatureCounts.wdl" target="_blank" rel="noopener noreferrer">CountAlignments.CountAlignments</a></td><td>Uses a custom GTF with featureCounts and Python to mark introns, create a BAM that has alignments spanning intron-exon junctions removed, and counts exons using the custom BAM and by excluding intron tags.</td><td><a href="http://subread.sourceforge.net/" target="_blank" rel="noopener noreferrer">Subread</a></td><td><a href="http://bioinf.wehi.edu.au/featureCounts/" target="_blank" rel="noopener noreferrer">FeatureCounts</a>, Python 3</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/skylab/LoomUtils.wdl" target="_blank" rel="noopener noreferrer">LoomUtils.SingleNucleusSmartSeq2LoomOutput</a></td><td>Creates the matrix files (Loom format) for each sample.</td><td>Python 3</td><td>Custom script: <a href="https://github.com/broadinstitute/warp/blob/develop/dockers/skylab/loom-output/ss2_loom_merge.py" target="_blank" rel="noopener noreferrer">ss2_loom_merge.py</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/skylab/LoomUtils.wdl" target="_blank" rel="noopener noreferrer">LoomUtils.AggregateSmartSeq2Loom</a></td><td>Aggregates the matrix files (Loom format) for each sample to produce one final Loom output.</td><td>Python 3</td><td>Custom script: <a href="https://github.com/broadinstitute/warp/blob/develop/dockers/skylab/loom-output/ss2_loom_merge.py" target="_blank" rel="noopener noreferrer">ss2_loom_merge.py</a></td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="trimming-adapters"></a>Trimming adapters<a class="hash-link" href="#trimming-adapters" title="Direct link to heading">#</a></h3><p>The TrimAdapters task uses the adapter list reference file to run the <a href="https://github.com/ExpressionAnalysis/ea-utils/tree/master/clipper" target="_blank" rel="noopener noreferrer">fastq-mcf</a> tool. This tool identifies the adapters in the input FASTQ files and performs clipping by using a subsampling parameter of 200,000 reads. The task outputs the trimmed FASTQ files which are then used for alignment. </p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="aligning-reads"></a>Aligning reads<a class="hash-link" href="#aligning-reads" title="Direct link to heading">#</a></h3><p>The StarAlignFastq task runs the STAR aligner on the trimmed FASTQ files. The STAR quantMode parameter is set to GeneCounts, which counts the number of reads per gene while mapping. The task outputs a coordinate-sorted aligned BAM file. </p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="removing-duplicate-reads"></a>Removing duplicate reads<a class="hash-link" href="#removing-duplicate-reads" title="Direct link to heading">#</a></h3><p>The RemoveDuplicatesFromBam task removes multi-mapped reads, optical duplicates, and PCR duplicates from the aligned BAM. It then adds readgroup information and creates a new, coordinate-sorted aligned BAM output.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="collecting-metrics"></a>Collecting metrics<a class="hash-link" href="#collecting-metrics" title="Direct link to heading">#</a></h3><p>The CollectMultipleMetrics task uses the Picard tool CollectMultipleMetrics to perform QC on the deduplicated BAM file. These metrics are copied to the final cell-by-gene matrix output (Loom file).</p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="counting-genes"></a>Counting genes<a class="hash-link" href="#counting-genes" title="Direct link to heading">#</a></h3><p>The CountAlignments task uses the featureCounts package to count introns and exons. First, the featureCounts tool counts intronic alignments in the deduplicated BAM using a custom GTF with annotated introns. The tool flags intronic alignments if they overlap an annotated intron by a minimum of 3 bp. </p><p>Next, following pipeline processes established by the BICCN Whole Mouse Brain Working Group, a custom Python script (“<a href="https://github.com/broadinstitute/warp/tree/develop/dockers/skylab/featureCounts/remove-reads-on-junctions.py" target="_blank" rel="noopener noreferrer">remove-reads-on-junctions.py</a>”) removes all alignments in the deduplicated BAM that cross only one intron-exon junction and produces an intermediate BAM file for exon counting. This removes a negligible amount of putative intronic alignments that did not meet the 3 bp intron overlap criteria. </p><p>Lastly, featureCounts uses the intermediate BAM with junctions removed to count exons. The final outputs of this step include a cell-by-gene matrix of intronic counts, a cell-by-gene matrix of exonic counts, and summary metrics for the different count types.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="creating-the-cell-by-gene-matrix-loom"></a>Creating the cell-by-gene matrix (Loom)<a class="hash-link" href="#creating-the-cell-by-gene-matrix-loom" title="Direct link to heading">#</a></h3><p>The LoomUtils task combines the Picard metrics (alignment_summary_metrics, deduplication metrics, and the G/C bias summary metrics) with the featureCount exon and intron counts to create a Loom formatted cell-by-gene count matrix. </p><p>Exonic counts are in the Loom matrix and intronic counts are added as a Loom layer. Read more about Loom file format in the <a href="https://linnarssonlab.org/loompy/index.html" target="_blank" rel="noopener noreferrer">Loompy documentation</a>.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="outputs"></a>Outputs<a class="hash-link" href="#outputs" title="Direct link to heading">#</a></h3><p>The table below details the final outputs of the Multi-snSS2 workflow.  </p><table><thead><tr><th>Output Name</th><th>Output Description</th><th>Output Format</th></tr></thead><tbody><tr><td>loom_output</td><td>Cell-by-gene count matrix that includes the raw exon counts (in matrix), intron counts (in matrix layer), cell metrics (column attributes) and gene IDs (row attributes).</td><td>Loom</td></tr><tr><td>bam_files</td><td>Array of genome-aligned BAM files (one for each cell) generated with Star.</td><td>Array [BAM]</td></tr><tr><td>exon_intron_count_files</td><td>Array of TXT files (one per cell) that contain intronic and exonic counts.</td><td>Array [TXT]</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="validation"></a>Validation<a class="hash-link" href="#validation" title="Direct link to heading">#</a></h2><p>The Multi-snSS2 pipeline was scientifically validated by the BRAIN Initiatives Cell Census Network (BICCN) 2.0 Whole Mouse Brain Working Group. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="versioning"></a>Versioning<a class="hash-link" href="#versioning" title="Direct link to heading">#</a></h2><p>All Multi-snSS2 release notes are documented in the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/smartseq2_single_nucleus_multisample/MultiSampleSmartSeq2SingleNucleus.changelog.md" target="_blank" rel="noopener noreferrer">Multi-snSS2 changelog</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="citing-the-multi-snss2-pipeline"></a>Citing the Multi-snSS2 Pipeline<a class="hash-link" href="#citing-the-multi-snss2-pipeline" title="Direct link to heading">#</a></h2><p>To cite the Multi-snSS2 pipeline, use the <a href="https://scicrunch.org/scicrunch/Resources/record/nlx_144509-1/SCR_021312/resolver" target="_blank" rel="noopener noreferrer">SciCrunch resource identifier</a>.</p><ul><li>Ex: <em>Smart-seq2 Single Nucleus Multi-Sample Pipeline (RRID:SCR_021312)</em></li></ul><p>To view an example of this citation as well as a publication-style methods section, see the Multi-snSS2 <a href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/multi_snss2.methods">Example Methods</a> </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="consortia-support"></a>Consortia Support<a class="hash-link" href="#consortia-support" title="Direct link to heading">#</a></h2><p>This pipeline is supported and used by the <a href="https://biccn.org/" target="_blank" rel="noopener noreferrer">BRAIN Initiative Cell Census Network</a> (BICCN). </p><p>If your organization also uses this pipeline, we would love to list you! Please reach out to us by contacting <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="have-suggestions"></a>Have Suggestions?<a class="hash-link" href="#have-suggestions" title="Direct link to heading">#</a></h2><p>Help us make our tools better by contacting <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a> for pipeline-related suggestions or questions.</p></div><footer class="docusaurus-mt-lg"><div class="row"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/README.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_">Last updated on <b><time datetime="2021-10-06T18:50:18.000Z">10/6/2021</time></b> by <b>Denis Loginov</b></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/Loom_schema"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">« What&#x27;s in the Smart-seq2 Multi Sample Pipeline Loom File?</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/multi_snss2.methods"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Smart-seq2 Single Nucleus Multi-Sample v1.1.0 Publication Methods »</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#introduction-to-the-smart-seq2-single-nucleus-multi-sample-pipeline" class="table-of-contents__link">Introduction to the Smart-seq2 Single Nucleus Multi-Sample pipeline</a></li><li><a href="#quick-start-table" class="table-of-contents__link">Quick start table</a></li><li><a href="#set-up" class="table-of-contents__link">Set-Up</a><ul><li><a href="#multi-snss2-installation-and-requirements" class="table-of-contents__link">Multi-snSS2 installation and requirements</a></li><li><a href="#inputs" class="table-of-contents__link">Inputs</a></li></ul></li><li><a href="#running-multi-snss2" class="table-of-contents__link">Running Multi-snSS2</a><ul><li><a href="#multi-snss2-workflow-summary" class="table-of-contents__link">Multi-snSS2 workflow summary</a></li><li><a href="#trimming-adapters" class="table-of-contents__link">Trimming adapters</a></li><li><a href="#aligning-reads" class="table-of-contents__link">Aligning reads</a></li><li><a href="#removing-duplicate-reads" class="table-of-contents__link">Removing duplicate reads</a></li><li><a href="#collecting-metrics" class="table-of-contents__link">Collecting metrics</a></li><li><a href="#counting-genes" class="table-of-contents__link">Counting genes</a></li><li><a href="#creating-the-cell-by-gene-matrix-loom" class="table-of-contents__link">Creating the cell-by-gene matrix (Loom)</a></li><li><a href="#outputs" class="table-of-contents__link">Outputs</a></li></ul></li><li><a href="#validation" class="table-of-contents__link">Validation</a></li><li><a href="#versioning" class="table-of-contents__link">Versioning</a></li><li><a href="#citing-the-multi-snss2-pipeline" class="table-of-contents__link">Citing the Multi-snSS2 Pipeline</a></li><li><a href="#consortia-support" class="table-of-contents__link">Consortia Support</a></li><li><a href="#have-suggestions" class="table-of-contents__link">Have Suggestions?</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2021 Copyright © Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.90c71525.js"></script>
<script src="/warp/assets/js/main.1fa25a74.js"></script>
</body>
</html>