<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.5">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">Smart-seq2 Single Sample Overview | WARP</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Smart-seq2 Single Sample Overview | WARP"><meta data-react-helmet="true" name="description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><meta data-react-helmet="true" property="og:description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.57c5b218.css">
<link rel="preload" href="/warp/assets/js/runtime~main.90c71525.js" as="script">
<link rel="preload" href="/warp/assets/js/main.1fa25a74.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_3Zt9 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">ðŸŒž</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button class="clean-btn backToTopButton_35hR" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed menuWithAnnouncementBar_2WvA"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">About WARP</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">WARP Pipelines</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">CEMBA</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Exome Germline Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">GDC Whole Genome Somatic Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Illumina Genotyping Array</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Imputation</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Optimus</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Single Cell ATAC</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">Smart-seq2 Single Sample</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README">Smart-seq2 Single Sample Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/Loom_schema">What&#x27;s in the Smart-seq2 Single Sample Pipeline Loom File?</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Whole Genome Germline Single Sample</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Contribution Guide</a></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_1CGd"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_3E-R"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="tocCollapsible_1PrD tocMobile_3Hoh"><button type="button" class="clean-btn tocCollapsibleButton_2O1e">On this page</button></div><div class="markdown"><header><h1>Smart-seq2 Single Sample Overview</h1></header><table><thead><tr><th align="center">Pipeline Version</th><th align="center">Date Updated</th><th align="center">Documentation Author</th><th align="center">Questions or Feedback</th></tr></thead><tbody><tr><td align="center"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">smartseq2_v5.1.1</a></td><td align="center">December, 2020</td><td align="center"><a href="mailto:ekiernan@broadinstitute.org" target="_blank" rel="noopener noreferrer">Elizabeth Kiernan</a></td><td align="center">Please file GitHub issues in WARP or contact <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a></td></tr></tbody></table><p><img src="/warp/assets/images/smartseq_image-4574ae53d1ef76e202eac615f71940a8.png"></p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="introduction-to-the-smart-seq2-single-sample-pipeline"></a>Introduction to the Smart-seq2 Single Sample Pipeline<a class="hash-link" href="#introduction-to-the-smart-seq2-single-sample-pipeline" title="Direct link to heading">#</a></h2><p>The Smart-seq2 Single Sample Pipeline (SS2) is designed by the <a href="https://data.humancellatlas.org/about" target="_blank" rel="noopener noreferrer">Data Coordination Platform</a> of the <a href="https://www.humancellatlas.org/" target="_blank" rel="noopener noreferrer">Human Cell Atlas</a> to process single-cell RNAseq (scRNAseq) data generated by <a href="https://www.nature.com/articles/nmeth.2639" target="_blank" rel="noopener noreferrer">Smart-seq2 assays</a>. The pipeline&#x27;s workflow is written in WDL, is freely available on <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl" target="_blank" rel="noopener noreferrer">GitHub</a>, and can be run by any compliant WDL runner (e.g. <a href="https://github.com/broadinstitute/cromwell" target="_blank" rel="noopener noreferrer">Crowmell</a>). The pipeline is designed to process scRNA-seq data from an individual cell (to process multiple cells, see the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_multisample/MultiSampleSmartSeq2.wdl" target="_blank" rel="noopener noreferrer">Smart-seq2_Multi Sample</a> pipeline).</p><p>The SS2 workflow is divided into two parts which run in parallel. In part one, the workflow aligns reads to the genome and performs quality control on genome-aligned BAMs. In part two, it aligns reads to the transcriptome and generates gene expression estimates from transcriptome-aligned BAMs. The pipeline returns reads and expression estimates in BAM, and read counts and QC metrics in Loom and CSV file format.</p><p>The pipeline has been validated to support <a href="https://docs.google.com/document/d/1MonsTG8UnROHZ_XpulrSZNTxO988KEH6T6h45plFYQg/edit#heading=h.ixoqmhbabdvh" target="_blank" rel="noopener noreferrer">human</a> and <a href="https://docs.google.com/document/d/12zGTFROrcXEByt9z0h06qjSqb9vWutn28Tx6YiND1Ds/edit" target="_blank" rel="noopener noreferrer">mouse</a>, stranded or unstranded, paired- or single-end, and plate- or fluidigm-based Smart-seq2 datasets. Read more in the <a href="#validation">validation section</a>.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Want to use the Smart-seq2 pipeline for your publication?</h5></div><div class="admonition-content"><p>Check out the <a href="/warp/docs/Pipelines/Smart-seq2_Multi_Sample_Pipeline/smart-seq2.methods">Smart-seq2 Publication Methods</a> to get started!</p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="quick-start-table"></a>Quick Start Table<a class="hash-link" href="#quick-start-table" title="Direct link to heading">#</a></h2><table><thead><tr><th>Pipeline Features</th><th>Description</th><th>Source</th></tr></thead><tbody><tr><td>Assay Type</td><td>Smart-seq2 Single Sample</td><td><a href="https://www.nature.com/articles/nprot.2014.006" target="_blank" rel="noopener noreferrer">Smart-seq2</a></td></tr><tr><td>Overall Workflow</td><td>Quality control and transcriptome quantification</td><td>Code available from <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl" target="_blank" rel="noopener noreferrer">GitHub</a></td></tr><tr><td>Workflow Language</td><td>WDL</td><td><a href="https://github.com/openwdl/wdl" target="_blank" rel="noopener noreferrer">openWDL</a></td></tr><tr><td>Genomic Reference Sequence (for validation)</td><td>GRCh38 human genome primary sequence and M21 (GRCm38.p6) mouse genome primary sequence</td><td>GENCODE <a href="https://www.gencodegenes.org/human/release_27.html" target="_blank" rel="noopener noreferrer">Human</a> and <a href="https://www.gencodegenes.org/mouse/release_M21.html" target="_blank" rel="noopener noreferrer">Mouse</a></td></tr><tr><td>Transcriptomic Reference Annotation (for validation)</td><td>V27 GENCODE human transcriptome and M21 mouse transcriptome</td><td>GENCODE <a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.annotation.gtf.gz" target="_blank" rel="noopener noreferrer">Human</a> and <a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M21/gencode.vM21.annotation.gff3.gz" target="_blank" rel="noopener noreferrer">Mouse</a></td></tr><tr><td>Aligner</td><td>HISAT2 (v.2.1.0)</td><td><a href="https://www.nature.com/articles/s41587-019-0201-4" target="_blank" rel="noopener noreferrer">Kim, et al.,2019</a></td></tr><tr><td>QC Metrics</td><td>Picard (v.2.10.10)</td><td><a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Broad Institute</a></td></tr><tr><td>Transcript Quantification</td><td>Utilities for processing large-scale single cell datasets</td><td><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323" target="_blank" rel="noopener noreferrer">RSEM v.1.3.0</a></td></tr><tr><td>Data Input File Format</td><td>File format in which sequencing data is provided</td><td><a href="https://academic.oup.com/nar/article/38/6/1767/3112533" target="_blank" rel="noopener noreferrer">FASTQ</a></td></tr><tr><td>Data Output File Formats</td><td>File formats in which Smart-seq2 output is provided</td><td><a href="http://samtools.github.io/hts-specs/" target="_blank" rel="noopener noreferrer">BAM</a>, Loom (generated with <a href="http://loompy.org/" target="_blank" rel="noopener noreferrer">Loompy v.3.0.6)</a>, CSV (QC metrics and counts)</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="set-up"></a>Set-Up<a class="hash-link" href="#set-up" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="smart-seq2-single-sample-installation-and-requirements"></a>Smart-seq2 Single Sample Installation and Requirements<a class="hash-link" href="#smart-seq2-single-sample-installation-and-requirements" title="Direct link to heading">#</a></h3><p>The SS2 workflow code can be downloaded by cloning the GitHub <a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer">WARP repository</a>. For the latest release of Smart-seq2, please see the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.changelog.md" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Sample changelog</a>.</p><p>The workflow is deployed using <a href="https://github.com/broadinstitute/cromwell" target="_blank" rel="noopener noreferrer">Cromwell</a>, a GA4GH compliant, flexible workflow management system that supports multiple computing platforms.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Try the SS2 workflow in Terra</h5></div><div class="admonition-content"><p>You can run the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_multisample/MultiSampleSmartSeq2.wdl" target="_blank" rel="noopener noreferrer">Smart-seq2 Multi-Sample workflow</a> (a wrapper for the Single Sample workflow described in this document) in <a href="https://app.terra.bio" target="_blank" rel="noopener noreferrer">Terra</a>, a cloud-based analysis platform. The Terra <a href="https://app.terra.bio/#workspaces/featured-workspaces-hca/HCA%20Smart-seq2%20Multi%20Sample%20Pipeline" target="_blank" rel="noopener noreferrer">Smart-seq2 public workspace</a> is preloaded with the Smart-seq2 workflow, example testing data, and all the required reference data.</p></div></div><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="inputs"></a>Inputs<a class="hash-link" href="#inputs" title="Direct link to heading">#</a></h3><p>There are two example configuration (JSON) files available to test the SS2 workflow.</p><ul><li><a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/human_single_example.json" target="_blank" rel="noopener noreferrer">human_single_example.json</a>: Configurations for an example single-end human dataset</li><li><a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample/mouse_paired_example.json" target="_blank" rel="noopener noreferrer">mouse_paired_example.json</a>: Configurations for an example mouse paired-end dataset</li></ul><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="sample-data-input"></a>Sample Data Input<a class="hash-link" href="#sample-data-input" title="Direct link to heading">#</a></h4><p>The pipeline is designed for both single- and paired-end reads in the form of FASTQ files. The workflow accepts two FASTQ files for paired-end experiments and one FASTQ file single-end experiments. It processes one sample (cell).</p><ul><li><code>fastq1</code>: forward reads for sample with paired-end sequencing (or reads for sample with single-end sequencing)</li><li><code>fastq2</code>: reverse reads for sample with paired-end sequencing (not applicable for samples with single-end sequencing)</li></ul><p>The workflow will use modified tasks depending if a sample is paired-end or single-end. This is designated with a boolean, as detailed in the following Reference Inputs section.</p><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="additional-reference-inputs"></a>Additional Reference Inputs<a class="hash-link" href="#additional-reference-inputs" title="Direct link to heading">#</a></h4><p>The Smart-seq2 Single Sample workflow requires multiple reference indexes. Information on how these indexes are created is found in the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/build_indices/README.md" target="_blank" rel="noopener noreferrer">BuildIndices workflow overview</a> documentation. The table below describes the references and each tool that uses the reference as input, when applicable.</p><table><thead><tr><th>Workflow Step</th><th>Reference name</th><th>Reference Description</th><th>Tool</th></tr></thead><tbody><tr><td>All</td><td>input_id</td><td>String containing unique identifier or name for the cell. It can be a human-readable descriptor (i.e. &quot;cell1&quot;) or a UUID</td><td>NA</td></tr><tr><td>All</td><td>input_name</td><td>Optional string that can be used as a name/identifier for the cell</td><td>NA</td></tr><tr><td>All</td><td>output_name</td><td>Output name, can include path</td><td>NA</td></tr><tr><td>All</td><td>paired_end</td><td>A boolean describing if sample is paired-end</td><td>NA</td></tr><tr><td>All</td><td>input_id_metadata_field</td><td>Optional string that describes, when applicable, the metadata field used for <code>input_id</code></td><td>NA</td></tr><tr><td>All</td><td>input_name_metadata_field</td><td>Optional string that describes, when applicable, the metadata field used for the <code>input_name</code></td><td>NA</td></tr><tr><td><strong>Genomic alignment with HISAT2</strong></td><td>hisat2_ref_index</td><td>HISAT2 reference index file in tarball</td><td>HISAT2</td></tr><tr><td></td><td>hisat2_ref_name</td><td>HISAT2 reference index name</td><td>HISAT2</td></tr><tr><td><strong>Picard-generated quality control metrics</strong></td><td>genome_ref_fasta</td><td>Genome reference in fasta format</td><td>Picard</td></tr><tr><td></td><td>gene_ref_flat</td><td><a href="https://software.broadinstitute.org/software/igv/genePred" target="_blank" rel="noopener noreferrer">RefFlat</a> file containing the location of RNA transcripts, exon start sites, etc.</td><td>Picard</td></tr><tr><td></td><td>rrna_intervals</td><td>Ribosomal RNA intervals file</td><td>Picard</td></tr><tr><td></td><td>stranded</td><td>Library strand information for HISAT2; example values include FR (read corresponds to transcript), RF(read corresponds to reverse complement of transcript), or NONE</td><td>Picard</td></tr><tr><td><strong>Transcriptomic alignment with HISAT2</strong></td><td>hisat2_ref_trans_index</td><td>HISAT2 transcriptome index file in tarball</td><td>HISAT2</td></tr><tr><td></td><td>hisat2_ref_trans_name</td><td>HISAT2 transcriptome index file name</td><td>HISAT2</td></tr><tr><td><strong>Gene expression quantification with RSEM</strong></td><td>rsem_ref_index</td><td>RSEM reference index file in tarball</td><td>RSEM</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="running-smart-seq2"></a>Running Smart-seq2<a class="hash-link" href="#running-smart-seq2" title="Direct link to heading">#</a></h2><p>The <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl" target="_blank" rel="noopener noreferrer">SmartSeq2SingleSample.wdl</a> is in the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_sample" target="_blank" rel="noopener noreferrer">pipelines/smartseq2_single_sample folder</a> of the WARP repository and implements the workflow by importing individual tasks (written in WDL script) from the WARP <a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/" target="_blank" rel="noopener noreferrer">tasks folder</a>.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="smart-seq2-workflow-summary"></a>Smart-seq2 Workflow Summary<a class="hash-link" href="#smart-seq2-workflow-summary" title="Direct link to heading">#</a></h3><p>Overall, the workflow is divided into two parts that are completed after an initial input validation step. Each workflow part comprises tasks (WDL scripts) that are summarized below and are available in the WARP <a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/" target="_blank" rel="noopener noreferrer">tasks folder</a> in GitHub.</p><p><strong>Part 1: Quality Control Tasks</strong></p><ol><li>Aligns reads to the genome with HISAT2 v.2.1.0</li><li>Calculates summary metrics from an aligned BAM using Picard v.2.10.10</li></ol><p><strong>Part 2: Transcriptome Quantification Tasks</strong></p><ol><li>Aligns reads to the transcriptome with HISAT v.2.1.0</li><li>Quantifies gene expression using RSEM v.1.3.0</li></ol><p>The table below provides links to each of the Smart-seq2 Single Sample pipeline&#x27;s task WDL scripts, as well as documentation for each task&#x27;s software tools.</p><p>If you are looking for the parameters used for each task/tool, click on the task link in the table below and see the <code>command {}</code> section of the task WDL. The task&#x27;s Docker image is also specified in the task WDL in the <code># runtime values</code> section as <code>String docker =</code>.</p><table><thead><tr><th>Task</th><th>Tool</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/HISAT2.wdl" target="_blank" rel="noopener noreferrer">HISAT2.HISATPairedEnd</a></td><td><a href="https://www.nature.com/articles/s41587-019-0201-4" target="_blank" rel="noopener noreferrer">HISAT2 software</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/HISAT2.wdl" target="_blank" rel="noopener noreferrer">HISAT2.HISATSingleEnd</a></td><td><a href="https://www.nature.com/articles/s41587-019-0201-4" target="_blank" rel="noopener noreferrer">HISAT2 software</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/Picard.wdl" target="_blank" rel="noopener noreferrer">Picard.CollectMultipleMetrics</a></td><td><a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Picard software</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/Picard.wdl" target="_blank" rel="noopener noreferrer">Picard.CollectRnaMetrics</a></td><td><a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Picard software</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/Picard.wdl" target="_blank" rel="noopener noreferrer">Picard.CollectDuplicationMetrics</a></td><td><a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Picard software</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/HISAT2.wdl" target="_blank" rel="noopener noreferrer">HISAT2.HISAT2RSEM</a></td><td><a href="https://www.nature.com/articles/s41587-019-0201-4" target="_blank" rel="noopener noreferrer">HISAT2 software</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/HISAT2.wdl" target="_blank" rel="noopener noreferrer">HISAT2.HISAT2RSEMSingleEnd</a></td><td><a href="https://www.nature.com/articles/s41587-019-0201-4" target="_blank" rel="noopener noreferrer">HISAT2 software</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/RSEM.wdl" target="_blank" rel="noopener noreferrer">RSEM.RSEMExpression</a></td><td><a href="https://deweylab.github.io/RSEM/README.html#de" target="_blank" rel="noopener noreferrer">RSEM software</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/GroupMetricsOutputs.wdl" target="_blank" rel="noopener noreferrer">GroupMetricsOutputs.GroupQCOutputs</a></td><td><a href="https://sctools.readthedocs.io/en/latest/readme.html" target="_blank" rel="noopener noreferrer">sctools software</a></td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/LoomUtils.wdl" target="_blank" rel="noopener noreferrer">LoomUtils.SmartSeq2LoomOutput</a></td><td><a href="https://www.python.org/download/releases/3.0/" target="_blank" rel="noopener noreferrer">python3 software</a></td></tr></tbody></table><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="part-1-quality-control-tasks"></a>Part 1: Quality Control Tasks<a class="hash-link" href="#part-1-quality-control-tasks" title="Direct link to heading">#</a></h4><h5><a aria-hidden="true" tabindex="-1" class="anchor anchor__h5 anchorWithStickyNavbar_31ik" id="11-align-reads-to-the-genome-using-hisat2"></a>1.1 Align reads to the genome using HISAT2<a class="hash-link" href="#11-align-reads-to-the-genome-using-hisat2" title="Direct link to heading">#</a></h5><p>HISAT2 is a fast, cost-efficient alignment tool that can determine the presence of non-transcript sequences and true transcript sequences, taking into account the presence of single-nucleotide polymorphisms (<a href="https://www.nature.com/articles/s41587-019-0201-4" target="_blank" rel="noopener noreferrer">Kim et al.,2019</a>). The Smart-seq2 Single Sample workflow uses the <a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/HISAT2.wdl" target="_blank" rel="noopener noreferrer">HISAT2 task</a> to call HISAT2 and perform graph-based alignment of paired- or single-end reads (in the form of FASTQ files) to a reference genome. This task requires a reference index which can be built using the <a href="https://github.com/broadinstitute/warp/tree/master/pipelines/skylab/build_indices" target="_blank" rel="noopener noreferrer">BuildIndices.wdl</a> documentation. The outputs of the task include a genome-aligned BAM file, a BAM index, and an alignment log file.</p><h5><a aria-hidden="true" tabindex="-1" class="anchor anchor__h5 anchorWithStickyNavbar_31ik" id="12-calculate-summary-metrics-using-picard"></a>1.2 Calculate summary metrics using Picard<a class="hash-link" href="#12-calculate-summary-metrics-using-picard" title="Direct link to heading">#</a></h5><p><a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Picard</a> is a suite of command line tools used for manipulating high-throughput sequencing data. The <a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/Picard.wdl" target="_blank" rel="noopener noreferrer">Picard task</a> uses Picard tools to calculate quality control measurements on the HISAT2 genome-aligned BAM file. The task requires a reference genome fasta, a RefFlat gene annotation file, and an RNA intervals file (see the <a href="https://github.com/broadinstitute/warp/tree/master/pipelines/skylab/build_indices" target="_blank" rel="noopener noreferrer">BuildIndices workflow</a> for creating references). The outputs of the task are text and PDF files for each metric.</p><p>The <a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/Picard.wdl" target="_blank" rel="noopener noreferrer">Picard task</a> generates QC metrics by using three sub-tasks:</p><ul><li><p>CollectMultipleMetrics: calls the  <a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/4.0.0.0/picard_analysis_CollectMultipleMetrics.php" target="_blank" rel="noopener noreferrer">CollectMultipleMetrics</a> tool which uses the aligned BAM file and reference genome fasta to collect metrics on <a href="http://broadinstitute.github.io/picard/picard-metric-definitions.html#AlignmentSummaryMetrics" target="_blank" rel="noopener noreferrer">alignment</a>, <a href="http://broadinstitute.github.io/picard/picard-metric-definitions.html#InsertSizeMetrics" target="_blank" rel="noopener noreferrer">insert size</a>, <a href="https://broadinstitute.github.io/picard/command-line-overview.html#CollectGcBiasMetrics" target="_blank" rel="noopener noreferrer">GC bias</a>, <a href="http://broadinstitute.github.io/picard/picard-metric-definitions.html#BaseDistributionByCycleMetrics" target="_blank" rel="noopener noreferrer">base distribution by cycle</a>, <a href="https://broadinstitute.github.io/picard/command-line-overview.html#QualityScoreDistribution" target="_blank" rel="noopener noreferrer">quality score distribution</a>, <a href="https://broadinstitute.github.io/picard/command-line-overview.html#MeanQualityByCycle" target="_blank" rel="noopener noreferrer">quality distribution by cycle</a>, <a href="http://broadinstitute.github.io/picard/picard-metric-definitions.html#ErrorSummaryMetrics" target="_blank" rel="noopener noreferrer">sequencing artifacts</a>, and <a href="http://broadinstitute.github.io/picard/picard-metric-definitions.html#CollectQualityYieldMetrics.QualityYieldMetrics" target="_blank" rel="noopener noreferrer">quality yield</a>.</p></li><li><p>CollectRnaMetrics: calls the <a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/4.0.0.0/picard_analysis_CollectRnaSeqMetrics.php" target="_blank" rel="noopener noreferrer">CollectRnaSeqMetrics</a> tool which uses the aligned BAM, a RefFlat genome annotation file, and a ribosomal intervals file to produce RNA alignment metrics (metric descriptions are found in the <a href="http://broadinstitute.github.io/picard/picard-metric-definitions.html#RnaSeqMetrics" target="_blank" rel="noopener noreferrer">Picard Metrics Dictionary</a>).</p></li><li><p>CollectDuplicationMetrics: calls the <a href="https://software.broadinstitute.org/gatk/documentation/tooldocs/4.0.4.0/picard_sam_markduplicates_MarkDuplicates.php" target="_blank" rel="noopener noreferrer">MarkDuplicates</a> tool which uses the aligned BAM to identify duplicate reads (output metrics are listed in the <a href="http://broadinstitute.github.io/picard/picard-metric-definitions.html#DuplicationMetrics" target="_blank" rel="noopener noreferrer">Picard Metrics Dictionary</a>).</p></li></ul><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="part-2-transcriptome-quantification-tasks"></a>Part 2: Transcriptome Quantification Tasks<a class="hash-link" href="#part-2-transcriptome-quantification-tasks" title="Direct link to heading">#</a></h4><h5><a aria-hidden="true" tabindex="-1" class="anchor anchor__h5 anchorWithStickyNavbar_31ik" id="21-align-reads-to-the-transcriptome-using-hisat2"></a>2.1 Align reads to the transcriptome using HISAT2<a class="hash-link" href="#21-align-reads-to-the-transcriptome-using-hisat2" title="Direct link to heading">#</a></h5><p>The <a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/HISAT2.wdl" target="_blank" rel="noopener noreferrer">HISAT2.HISAT2RSEM task</a> uses HISAT2 to align reads to the reference transcriptome. The task requires the hisat2_ref_trans_index file and the sample FASTQ files as input. The output of this task is a transcriptome-aligned BAM file and an alignment log file.</p><h5><a aria-hidden="true" tabindex="-1" class="anchor anchor__h5 anchorWithStickyNavbar_31ik" id="22-quantify-gene-expression-using-rsem"></a>2.2 Quantify gene expression using RSEM<a class="hash-link" href="#22-quantify-gene-expression-using-rsem" title="Direct link to heading">#</a></h5><p><a href="https://deweylab.github.io/RSEM/README.html#de" target="_blank" rel="noopener noreferrer">RSEM</a> is a software package for quantifying transcript abundances (<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323" target="_blank" rel="noopener noreferrer">Li and Dewey, 2011</a>). The Smart-seq2 Single Sample workflow uses the <a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/RSEM.wdl" target="_blank" rel="noopener noreferrer">RSEM.RSEMExpression task</a> to calculate expression estimates from a transcriptome-aligned BAM file using the rsem_ref_index file for reference input. The RSEM tool <a href="http://deweylab.biostat.wisc.edu/rsem/rsem-calculate-expression.html" target="_blank" rel="noopener noreferrer">rsem-calculate-expression</a> is used to estimate gene/isoform expression.</p><p>The RSEM task returns the following output files:</p><ul><li>rsem_gene: gene level expression estimates in FPKM, TPM, and counts (&quot;expected_count&quot;)</li><li>rsem_isoform: isoform level expression estimates in FPKM, TPM, and counts (&quot;expected_count&quot;)</li><li>rsem_time: the time consumed by aligning reads</li><li>rsem_cnt: alignment statistics</li><li>rsem_model: RNA-seq model parameters</li><li>rsem_theta: fraction of reads resulting from background</li></ul><p>Only the rsem_gene (TPM and expected_count), rsem_isoform (TPM and expected_count), and rsem_cnt files are used for the final outputs of the Smart-seq2 Single Sample workflow.</p><p>After the Smart-seq2 workflow generates HISAT2, Picard and RSEM metrics, the <a href="https://github.com/broadinstitute/warp/blob/master/tasks/skylab/GroupMetricsOutputs.wdl" target="_blank" rel="noopener noreferrer">GroupMetricsOutputs task</a> combines most metrics into a &quot;group_results&quot; CSV file array. The workflow uses this array in downstream Loom file generation. The following HISAT2, <a href="http://broadinstitute.github.io/picard/picard-metric-definitions.html#" target="_blank" rel="noopener noreferrer">Picard</a> and <a href="http://deweylab.biostat.wisc.edu/rsem/rsem-calculate-expression.html" target="_blank" rel="noopener noreferrer">RSEM</a> outputs are inputs for this task:</p><ul><li>HISAT2_log_file</li><li>base_call_dist_metrics</li><li>gc_bias_detail_metrics</li><li>pre_adapter_details_metrics</li><li>pre_adapter_summary_metrics</li><li>bait_bias_detail_metrics</li><li>error_summary_metrics</li><li>alignment_summary_metrics</li><li>dedup_metrics</li><li>rna_metrics</li><li>gc_bias_summary_metrics</li><li>HISAT2RSEM_log_file</li><li>RSEMExpression.rsem_cnt</li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="outputs"></a>Outputs<a class="hash-link" href="#outputs" title="Direct link to heading">#</a></h3><p>The table below details the final outputs of the SS2 workflow.</p><table><thead><tr><th>Output Name</th><th>Output Description</th><th>Output Format</th></tr></thead><tbody><tr><td>Pipeline version</td><td>Version of the processing pipeline run on this data</td><td>string</td></tr><tr><td>aligned_bam</td><td>HISAT2 out BAM</td><td>BAM</td></tr><tr><td>bam_index</td><td>HISAT2 BAM index</td><td>BAM</td></tr><tr><td>insert_size_metrics</td><td>Picard insert size metrics</td><td>txt</td></tr><tr><td>quality_distribution_metrics</td><td>Picard quality distribution metrics</td><td>txt</td></tr><tr><td>quality_by_cycle_metrics</td><td>Picard quality by cycle metrics</td><td>txt</td></tr><tr><td>bait_bias_summary_metrics</td><td>Picard bait bias summary metrics</td><td>txt</td></tr><tr><td>rna_metrics</td><td>Picard RNA metrics</td><td>txt</td></tr><tr><td>group_results</td><td>Array of Picard metric files generated by the GroupQCOutputs task</td><td>CSV</td></tr><tr><td>rsem_gene_results</td><td>RSEM file containing gene-level expression estimates</td><td>tab-delimited</td></tr><tr><td>rsem_isoform_results</td><td>RSEM file containing isoform-level expression estimates</td><td>tab delimited</td></tr><tr><td>loom_output_files</td><td>Loom file containing RSEM TPM and expected_counts and metrics</td><td>Loom</td></tr></tbody></table><p>The final count matrix is in Loom format and includes only the group_results CSV files and the rsem_gene_results (TPM and expected_count). Note that the TPMs are contained in the Loom â€œmatrix&quot;, whereas expected_count is contained in the â€œlayers/estimated_counts&quot;.</p><div class="admonition admonition-warning alert alert--danger"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>Zarr Array Deprecation Notice June 2020</h5></div><div class="admonition-content"><p>Please note that we have deprecated the previously used Zarr array output. The pipeline now uses the Loom file format as the default output.</p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="validation"></a>Validation<a class="hash-link" href="#validation" title="Direct link to heading">#</a></h2><p>The SS2 pipeline has been validated for processing human and mouse, stranded or unstranded, paired- or single-end, and plate- or fluidigm-based Smart-seq2 datasets (see links to validation reports in the table below).</p><table><thead><tr><th>Workflow Configuration</th><th>Link to Report</th></tr></thead><tbody><tr><td>Mouse paired-end</td><td><a href="https://docs.google.com/document/d/12zGTFROrcXEByt9z0h06qjSqb9vWutn28Tx6YiND1Ds/edit" target="_blank" rel="noopener noreferrer">Report</a></td></tr><tr><td>Human and mouse single-end</td><td><a href="https://docs.google.com/document/d/1MonsTG8UnROHZ_XpulrSZNTxO988KEH6T6h45plFYQg/edit#heading=h.ixoqmhbabdvh" target="_blank" rel="noopener noreferrer">Report</a></td></tr><tr><td>Human stranded fluidigm</td><td><a href="https://docs.google.com/document/d/1FEg86Tlu657j9Kjw_v3keFQRXcBIs8gOqCwLbPSP-C0/edit#heading=h.wjr8otl7zg14" target="_blank" rel="noopener noreferrer">Report</a></td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="versioning"></a>Versioning<a class="hash-link" href="#versioning" title="Direct link to heading">#</a></h2><p>All SS2 release notes are documented in the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.changelog.md" target="_blank" rel="noopener noreferrer">Smartseq2 Single Sample changelog</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="-citing-the-smart-seq2-single-sample-pipeline"></a>## Citing the Smart-seq2 Single Sample Pipeline<a class="hash-link" href="#-citing-the-smart-seq2-single-sample-pipeline" title="Direct link to heading">#</a></h2><p>Please identify the SS2 pipeline in your methods section using the Smart-seq2 Single Sample Pipeline&#x27;s <a href="https://scicrunch.org/browse/resourcedashboard" target="_blank" rel="noopener noreferrer">SciCrunch resource identifier</a>.</p><ul><li>Ex: <em>Smart-seq2 Single Sample Pipeline (RRID:SCR_021228)</em></li></ul><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="consortia-support"></a>Consortia Support<a class="hash-link" href="#consortia-support" title="Direct link to heading">#</a></h2><p>This pipeline is supported and used by the <a href="https://www.humancellatlas.org/" target="_blank" rel="noopener noreferrer">Human Cell Atlas</a> (HCA) project. </p><p>If your organization also uses this pipeline, we would love to list you! Please reach out to us by contacting <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="have-suggestions"></a>Have Suggestions?<a class="hash-link" href="#have-suggestions" title="Direct link to heading">#</a></h2><p>Please help us make our tools better by contacting <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a> for pipeline-related suggestions or questions.</p></div><footer class="docusaurus-mt-lg"><div class="row"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/README.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_">Last updated on <b><time datetime="2021-10-06T18:50:18.000Z">10/6/2021</time></b> by <b>Denis Loginov</b></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/multi_snss2.methods"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« Smart-seq2 Single Nucleus Multi-Sample v1.1.0 Publication Methods</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/Loom_schema"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">What&#x27;s in the Smart-seq2 Single Sample Pipeline Loom File? Â»</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#introduction-to-the-smart-seq2-single-sample-pipeline" class="table-of-contents__link">Introduction to the Smart-seq2 Single Sample Pipeline</a></li><li><a href="#quick-start-table" class="table-of-contents__link">Quick Start Table</a></li><li><a href="#set-up" class="table-of-contents__link">Set-Up</a><ul><li><a href="#smart-seq2-single-sample-installation-and-requirements" class="table-of-contents__link">Smart-seq2 Single Sample Installation and Requirements</a></li><li><a href="#inputs" class="table-of-contents__link">Inputs</a></li></ul></li><li><a href="#running-smart-seq2" class="table-of-contents__link">Running Smart-seq2</a><ul><li><a href="#smart-seq2-workflow-summary" class="table-of-contents__link">Smart-seq2 Workflow Summary</a></li><li><a href="#outputs" class="table-of-contents__link">Outputs</a></li></ul></li><li><a href="#validation" class="table-of-contents__link">Validation</a></li><li><a href="#versioning" class="table-of-contents__link">Versioning</a></li><li><a href="#-citing-the-smart-seq2-single-sample-pipeline" class="table-of-contents__link">## Citing the Smart-seq2 Single Sample Pipeline</a></li><li><a href="#consortia-support" class="table-of-contents__link">Consortia Support</a></li><li><a href="#have-suggestions" class="table-of-contents__link">Have Suggestions?</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright Â© 2021 Copyright Â© Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.90c71525.js"></script>
<script src="/warp/assets/js/main.1fa25a74.js"></script>
</body>
</html>