<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.5">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">Optimus Count Matrix Overview | WARP</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/Optimus_Pipeline/Loom_schema"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Optimus Count Matrix Overview | WARP"><meta data-react-helmet="true" name="description" content="The Optimus pipeline&#x27;s default count matrix output is a Loom file, an HDF5 file generated using Loompy v.3.0.6."><meta data-react-helmet="true" property="og:description" content="The Optimus pipeline&#x27;s default count matrix output is a Loom file, an HDF5 file generated using Loompy v.3.0.6."><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/Optimus_Pipeline/Loom_schema"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Optimus_Pipeline/Loom_schema" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Optimus_Pipeline/Loom_schema" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.57c5b218.css">
<link rel="preload" href="/warp/assets/js/runtime~main.90c71525.js" as="script">
<link rel="preload" href="/warp/assets/js/main.1fa25a74.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_3Zt9 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">ðŸŒž</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button class="clean-btn backToTopButton_35hR" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed menuWithAnnouncementBar_2WvA"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">About WARP</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">WARP Pipelines</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">CEMBA</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Exome Germline Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">GDC Whole Genome Somatic Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Illumina Genotyping Array</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Imputation</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">Optimus</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Optimus_Pipeline/README">Optimus Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Optimus_Pipeline/optimus.methods">Optimus v5.0.0 Methods</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Optimus_Pipeline/Bam_tags">What tags are included in an Optimus BAM file?</a></li><li class="menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/warp/docs/Pipelines/Optimus_Pipeline/Loom_schema">Optimus Count Matrix Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Optimus_Pipeline/consortia-processing">Consortia Data Processing</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Single Cell ATAC</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Whole Genome Germline Single Sample</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Contribution Guide</a></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_1CGd"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_3E-R"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="tocCollapsible_1PrD tocMobile_3Hoh"><button type="button" class="clean-btn tocCollapsibleButton_2O1e">On this page</button></div><div class="markdown"><header><h1>Optimus Count Matrix Overview</h1></header><p>The Optimus pipeline&#x27;s default count matrix output is a Loom file, an HDF5 file generated using <a href="http://loompy.org/" target="_blank" rel="noopener noreferrer">Loompy v.3.0.6</a>. </p><p>The matrix contains global attributes detailing how counts were generated for the single-cell or single-nucleus parameters (<a href="#table-1-global-attributes">Table 1</a>). It additionally contains UMI-corrected counts as well as multiple metrics for both individual cells (the columns of the matrix; <a href="#table-2-column-attributes-cell-metrics">Table 2</a>) and individual genes (the rows of the matrix; <a href="#table-3-row-attributes-gene-metrics">Table 3</a>). </p><p>The tables below document these metrics, list which tools generate them, and define them.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Additional Matrix Processing for Consortia</h5></div><div class="admonition-content"><p>Loom files generated by Optimus for consortia, such as the Human Cell Atlas (HCA) or the BRAIN Initiative Cell Census Network (BICCN), may have additional processing steps. Read the <a href="/warp/docs/Pipelines/Optimus_Pipeline/consortia-processing#hca-data-coordination-platform-matrix-processing">Consortia Processing Overview</a> for details on consortia-specific matrix changes.</p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="table-1-global-attributes"></a>Table 1. Global Attributes<a class="hash-link" href="#table-1-global-attributes" title="Direct link to heading">#</a></h2><p>The global attributes in the Loom apply to the whole file, not any specific part. There are two global attributes for the Optimus Loom.</p><table><thead><tr><th align="left">Attribute</th><th align="left">Details</th></tr></thead><tbody><tr><td align="left">optimus_output_schema_version</td><td align="left">String with the loom file spec version</td></tr><tr><td align="left">expression_data_type</td><td align="left">String describing if the pipeline counts exonic or whole transcript (exonic and intronic) reads. For the single-cell mode (counting_mode = sc_rna), the value will be &quot;exonic&quot;; for the single-nucleus mode (counting_mode = sn_rna), the value will be &quot;whole_transcript&quot;</td></tr><tr><td align="left">input_id</td><td align="left">The sample or cell id listed in the pipeline configuration file. This can be any string, but we recommend it be consistent with any sample metadata.</td></tr><tr><td align="left">input_name</td><td align="left">Optional string that can be used to further describe the input</td></tr><tr><td align="left">input_id_metadata_field</td><td align="left">Optional string that describes, when applicable, the metadata field containing the <code>input_id</code></td></tr><tr><td align="left">input_name_metadata_field</td><td align="left">Optional string that describes, when applicable, the metadata field containing the <code>input_name</code></td></tr><tr><td align="left">pipeline_version</td><td align="left">String describing the Optimus version</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="table-2-column-attributes-cell-metrics"></a>Table 2. Column Attributes (Cell Metrics)<a class="hash-link" href="#table-2-column-attributes-cell-metrics" title="Direct link to heading">#</a></h2><table><thead><tr><th align="left">Cell Metrics</th><th align="center">Program</th><th align="left">Details</th></tr></thead><tbody><tr><td align="left"><code>CellID</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The unique identifier for each cell based on cell barcodes; identical to <code>cell_names</code>.</td></tr><tr><td align="left"><code>cell_names</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The unique identifier for each cell based on cell barcodes; identical to <code>CellID</code>.</td></tr><tr><td align="left"><code>input_id</code></td><td align="center">Provided as pipeline input</td><td align="left">The sample or cell id listed in the pipeline configuration file. This can be any string, but we recommend it be consistent with any sample metadata.</td></tr><tr><td align="left"><code>n_reads</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads associated with this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.n_reads" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>noise_reads</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">Number of reads that are categorized by 10x Genomics Cell Ranger as &quot;noise&quot;. Refers to long polymers, or reads with high numbers of N (ambiguous) nucleotides. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.noise_reads" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>perfect_molecule_barcodes</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads with molecule barcodes that have no errors. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.perfect_molecule_barcodes" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>n_mitochondrial_genes</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of mitochondrial genes detected by this cell. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.n_mitochondrial_genes" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>n_mitochondrial_molecules</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of molecules from mitochondrial genes detected for this cell. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.n_mitochondrial_molecules" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>pct_mitochondrial_molecules</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The percentage of molecules from mitochondrial genes detected for this cell. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.pct_mitochondrial_molecules" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>reads_mapped_exonic</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads for this entity that are mapped to exons. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_exonic" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>reads_mapped_intronic</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads for this entity that are mapped to introns. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_intronic" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>reads_mapped_utr</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads for this entity that are mapped to 3&#x27; untranslated regions (UTRs). <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_utr" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>reads_mapped_uniquely</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads mapped to a single unambiguous location in the genome. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_uniquely" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>reads_mapped_multiple</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads mapped to multiple genomic positions with equal confidence. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_multiple" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>duplicate_reads</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads that are duplicates (see README.md for definition of a duplicate). <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.duplicate_reads" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>spliced_reads</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads that overlap splicing junctions. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.spliced_reads" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>antisense_reads</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads that are mapped to the antisense strand instead of the transcribed strand. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.antisense_reads" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>molecule_barcode_fraction_bases_above_30_mean</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The average fraction of bases in molecule barcodes that receive quality scores greater than 30 across the reads of this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.molecule_barcode_fraction_bases_above_30_mean" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>molecule_barcode_fraction_bases_above_30_variance</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The variance in the fraction of bases in molecule barcodes that receive quality scores greater than 30 across the reads of this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.molecule_barcode_fraction_bases_above_30_variance" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>genomic_reads_fraction_bases_quality_above_30_mean</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The average fraction of bases in the genomic read that receive quality scores greater than 30 across the reads of this entity (included for 10x Cell Ranger count comparison). <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.genomic_reads_fraction_bases_quality_above_30_mean" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>genomic_reads_fraction_bases_quality_above_30_variance</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The variance in the fraction of bases in the genomic read that receive quality scores greater than 30 across the reads of this entity (included for 10x Cell Ranger count comparison). <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.genomic_reads_fraction_bases_quality_above_30_variance" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>genomic_read_quality_mean</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">Average quality of Illumina base calls in the genomic reads corresponding to this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.genomic_read_quality_mean" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>genomic_read_quality_variance</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">Variance in quality of Illumina base calls in the genomic reads corresponding to this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.genomic_read_quality_variance" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>n_molecules</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">Number of molecules corresponding to this entity. See README.md for the definition of a Molecule. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.n_molecules" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>n_fragments</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">Number of fragments corresponding to this entity. See README.md for the definition of a Fragment. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.n_fragments" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>reads_per_fragment</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The average number of reads associated with each fragment in this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_per_fragment" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>fragments_per_molecule</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The average number of fragments associated with each molecule in this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.fragments_per_molecule" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>fragments_with_single_read_evidence</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of fragments associated with this entity that are observed by only one read. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.fragments_with_single_read_evidence" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>molecules_with_single_read_evidence</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of molecules associated with this entity that are observed by only one read. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.molecules_with_single_read_evidence" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>perfect_cell_barcodes</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads whose cell barcodes contain no error. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.perfect_cell_barcodes" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>reads_mapped_intergenic</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads mapped to an intergenic region for this cell. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_intergenic" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>reads_mapped_too_many_loci</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of reads that were mapped to too many loci across the genome and as a consequence, are reported unmapped by the aligner. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_too_many_loci" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>cell_barcode_fraction_bases_above_30_variance</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The variance of the fraction of Illumina base calls for the cell barcode sequence that are greater than 30, across molecules. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.cell_barcode_fraction_bases_above_30_variance" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>cell_barcode_fraction_bases_above_30_mean</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The average fraction of Illumina base calls for the cell barcode sequences that are greater than 30, across molecules. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.cell_barcode_fraction_bases_above_30_mean" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>n_genes</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of genes detected by this cell. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.n_genes" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>genes_detected_multiple_observations</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">The number of genes that are observed by more than one read in this cell. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.genes_detected_multiple_observations" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td align="left"><code>reads_unmapped</code></td><td align="center"><a href="https://github.com/HumanCellAtlas/sctools/blob/master/src/sctools/metrics/aggregator.py" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td align="left">Reads that are non-transcriptomic</td></tr><tr><td align="left"><code>emptydrops_FDR</code></td><td align="center"><a href="https://bioconductor.org/packages/release/bioc/html/DropletUtils.html" target="_blank" rel="noopener noreferrer">dropletUtils</a></td><td align="left">False Discovery Rate (FDR) for being a non-empty droplet; single-cell data will read &quot;NA&quot; if task is unable to detect knee point inflection. Column is not included for data run in the sn_rna mode</td></tr><tr><td align="left"><code>emptydrops_IsCell</code></td><td align="center"><a href="https://bioconductor.org/packages/release/bioc/html/DropletUtils.html" target="_blank" rel="noopener noreferrer">dropletUtils</a></td><td align="left">Binarized call of cell/background based on predefined FDR cutoff; single-cell data will read &quot;NA&quot; if task is unable to detect knee point inflection. Column is not included for data run in the sn_rna mode</td></tr><tr><td align="left"><code>emptydrops_Limited</code></td><td align="center"><a href="https://bioconductor.org/packages/release/bioc/html/DropletUtils.html" target="_blank" rel="noopener noreferrer">dropletUtils</a></td><td align="left">Indicates whether a lower p-value could be obtained by increasing the number of iterations; single-cell data will read &quot;NA&quot; if task is unable to detect knee point inflection. Column is not included for data run in the sn_rna mode</td></tr><tr><td align="left"><code>emptydrops_LogProb</code></td><td align="center"><a href="https://bioconductor.org/packages/release/bioc/html/DropletUtils.html" target="_blank" rel="noopener noreferrer">dropletUtils</a></td><td align="left">The log-probability of observing the barcodeâ€™s count vector under the null model; single-cell data will read &quot;NA&quot; if the task is unable to detect knee point inflection. Column is not included for data run in the sn_rna mode</td></tr><tr><td align="left"><code>emptydrops_PValue</code></td><td align="center"><a href="https://bioconductor.org/packages/release/bioc/html/DropletUtils.html" target="_blank" rel="noopener noreferrer">dropletUtils</a></td><td align="left">Numeric, the Monte Carlo p-value against the null model; single-cell data will read &quot;NA&quot; if task is unable to detect knee point inflection. Column is not included for data run in the sn_rna mode</td></tr><tr><td align="left"><code>emptydrops_Total</code></td><td align="center"><a href="https://bioconductor.org/packages/release/bioc/html/DropletUtils.html" target="_blank" rel="noopener noreferrer">dropletUtils</a></td><td align="left">Numeric, the total read counts for each barcode; single-cell data will read &quot;NA&quot; if task is unable to detect knee point inflection. Column is not included for data run in the sn_rna mode</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="table-3-row-attributes-gene-metrics"></a>Table 3. Row Attributes (Gene Metrics)<a class="hash-link" href="#table-3-row-attributes-gene-metrics" title="Direct link to heading">#</a></h2><table><thead><tr><th>Gene Metrics</th><th>Program</th><th>Details</th></tr></thead><tbody><tr><td><code>ensembl_ids</code></td><td><a href="https://www.gencodegenes.org/" target="_blank" rel="noopener noreferrer">GENCODE GTF</a></td><td>The gene_id listed in the GENCODE GTF.</td></tr><tr><td><code>Gene</code></td><td><a href="https://www.gencodegenes.org/" target="_blank" rel="noopener noreferrer">GENCODE GTF</a></td><td>The unique gene_name provided in the GENCODE GTF; identical to the <code>gene_names</code> attribute.</td></tr><tr><td><code>gene_names</code></td><td><a href="https://www.gencodegenes.org/" target="_blank" rel="noopener noreferrer">GENCODE GTF</a></td><td>The unique gene_name provided in the GENCODE GTF; identical to the <code>Gene</code> attribute.</td></tr><tr><td><code>n_reads</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads associated with this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.n_reads" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>noise_reads</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads that are categorized by 10x Genomics Cell Ranger as &quot;noise&quot;. Refers to long polymers, or reads with high numbers of N (ambiguous) nucleotides. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.noise_reads" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>perfect_molecule_barcodes</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads with molecule barcodes that have no errors. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.perfect_molecule_barcodes" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>reads_mapped_exonic</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads for this entity that are mapped to exons. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_exonic" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>reads_mapped_intronic</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads for this entity that are mapped to introns. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_intronic" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>reads_mapped_utr</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads for this entity that are mapped to 3&#x27; untranslated regions (UTRs). <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_utr" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>reads_mapped_uniquely</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads mapped to a single unambiguous location in the genome. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_uniquely" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>reads_mapped_multiple</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads mapped to multiple genomic positions with equal confidence. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_mapped_multiple" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>duplicate_reads</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads that are duplicates (see README.md for definition of a duplicate).  <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.duplicate_reads" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>spliced_reads</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads that overlap splicing junctions. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.spliced_reads" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>antisense_reads</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of reads that are mapped to the antisense strand instead of the transcribed strand. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.antisense_reads" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>molecule_barcode_fraction_bases_above_30_mean</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The average fraction of bases in molecule barcodes that receive quality scores greater than 30 across the reads of this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.molecule_barcode_fraction_bases_above_30_mean" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>molecule_barcode_fraction_bases_above_30_variance</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The variance in the fraction of bases in molecule barcodes that receive quality scores greater than 30 across the reads of this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.molecule_barcode_fraction_bases_above_30_variance" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>genomic_reads_fraction_bases_quality_above_30_mean</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The average fraction of bases in the genomic read that receive quality scores greater than 30 across the reads of this entity (included for 10x Cell Ranger count comparison). <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.genomic_reads_fraction_bases_quality_above_30_mean" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>genomic_reads_fraction_bases_quality_above_30_variance</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The variance in the fraction of bases in the genomic read that receive quality scores greater than 30 across the reads of this entity (included for 10x Cell Ranger count comparison). <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.genomic_reads_fraction_bases_quality_above_30_variance" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>genomic_read_quality_mean</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>Average quality of Illumina base calls in the genomic reads corresponding to this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.genomic_read_quality_mean" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>genomic_read_quality_variance</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>Variance in quality of Illumina base calls in the genomic reads corresponding to this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.genomic_read_quality_variance" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>n_molecules</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>Number of molecules corresponding to this entity. See README.md for the definition of a Molecule. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.n_molecules" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>n_fragments</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>Number of fragments corresponding to this entity. See README.md for the definition of a Fragment. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.n_fragments" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>reads_per_molecule</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The average number of reads associated with each molecule in this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_per_molecule" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>reads_per_fragment</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The average number of reads associated with each fragment in this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.reads_per_fragment" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>fragments_per_molecule</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The average number of fragments associated with each molecule in this entity. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.fragments_per_molecule" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>fragments_with_single_read_evidence</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of fragments associated with this entity that are observed by only one read. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.fragments_with_single_read_evidence" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>molecules_with_single_read_evidence</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of molecules associated with this entity that are observed by only one read. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.CellMetrics.molecules_with_single_read_evidence" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>number_cells_detected_multiple</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of cells which observe more than one read of this gene. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.GeneMetrics.number_cells_detected_multiple" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr><tr><td><code>number_cells_expressing</code></td><td><a href="https://github.com/HumanCellAtlas/sctools/tree/master/src/sctools/metrics" target="_blank" rel="noopener noreferrer">SC Tools</a></td><td>The number of cells that detect this gene. <a href="https://sctools.readthedocs.io/en/latest/sctools.metrics.html#sctools.metrics.aggregator.GeneMetrics.number_cells_expressing" target="_blank" rel="noopener noreferrer">Metrics Definitions</a></td></tr></tbody></table></div><footer class="docusaurus-mt-lg"><div class="row"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Optimus_Pipeline/Loom_schema.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_">Last updated on <b><time datetime="2021-10-06T18:50:18.000Z">10/6/2021</time></b> by <b>Denis Loginov</b></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Optimus_Pipeline/Bam_tags"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« What tags are included in an Optimus BAM file?</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Optimus_Pipeline/consortia-processing"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Consortia Data Processing Â»</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#table-1-global-attributes" class="table-of-contents__link">Table 1. Global Attributes</a></li><li><a href="#table-2-column-attributes-cell-metrics" class="table-of-contents__link">Table 2. Column Attributes (Cell Metrics)</a></li><li><a href="#table-3-row-attributes-gene-metrics" class="table-of-contents__link">Table 3. Row Attributes (Gene Metrics)</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright Â© 2021 Copyright Â© Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.90c71525.js"></script>
<script src="/warp/assets/js/main.1fa25a74.js"></script>
</body>
</html>