<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.5">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">Genomic Data Commons (GDC) Whole Genome Somatic Single Sample Overview | WARP</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/Genomic_Data_Commons_Whole_Genome_Somatic/README"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Genomic Data Commons (GDC) Whole Genome Somatic Single Sample Overview | WARP"><meta data-react-helmet="true" name="description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><meta data-react-helmet="true" property="og:description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/Genomic_Data_Commons_Whole_Genome_Somatic/README"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Genomic_Data_Commons_Whole_Genome_Somatic/README" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Genomic_Data_Commons_Whole_Genome_Somatic/README" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.57c5b218.css">
<link rel="preload" href="/warp/assets/js/runtime~main.90c71525.js" as="script">
<link rel="preload" href="/warp/assets/js/main.1fa25a74.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_3Zt9 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">ðŸŒž</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button class="clean-btn backToTopButton_35hR" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed menuWithAnnouncementBar_2WvA"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">About WARP</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">WARP Pipelines</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">CEMBA</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Exome Germline Single Sample</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">GDC Whole Genome Somatic Single Sample</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/warp/docs/Pipelines/Genomic_Data_Commons_Whole_Genome_Somatic/README">Genomic Data Commons (GDC) Whole Genome Somatic Single Sample Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Illumina Genotyping Array</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Imputation</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Optimus</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Single Cell ATAC</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Whole Genome Germline Single Sample</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Contribution Guide</a></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_1CGd"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_3E-R"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="tocCollapsible_1PrD tocMobile_3Hoh"><button type="button" class="clean-btn tocCollapsibleButton_2O1e">On this page</button></div><div class="markdown"><header><h1>Genomic Data Commons (GDC) Whole Genome Somatic Single Sample Overview</h1></header><table><thead><tr><th align="center">Pipeline Version</th><th align="center">Date Updated</th><th align="center">Documentation Author</th><th align="center">Questions or Feedback</th></tr></thead><tbody><tr><td align="center"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">GDCWholeGenomeSomaticSingleSample_v1.0.1</a></td><td align="center">January, 2021</td><td align="center"><a href="mailto:ekiernan@broadinstitute.org" target="_blank" rel="noopener noreferrer">Elizabeth Kiernan</a></td><td align="center">Please file GitHub issues in warp or contact <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a></td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="introduction-to-the-gdc-whole-genome-somatic-single-sample-pipeline"></a>Introduction to the GDC Whole Genome Somatic Single Sample pipeline<a class="hash-link" href="#introduction-to-the-gdc-whole-genome-somatic-single-sample-pipeline" title="Direct link to heading">#</a></h2><p>The GDC Whole Genome Somatic Single Sample (abbreviated GDC here) pipeline is the alignment and preprocessing workflow for genomic data designed for the National Cancer Institute&#x27;s <a href="https://gdc.cancer.gov/about-gdc" target="_blank" rel="noopener noreferrer">Genomic Data Commons</a>.</p><p>A high-level overview of the pipeline in addition to tool parameters are available on the <a href="https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/" target="_blank" rel="noopener noreferrer">GDC Documentation site</a>.</p><p>Overall, the pipeline converts reads (CRAM or BAM) to FASTQ and (re)aligns them to the latest human reference genome (see the <a href="#gdc-reference-genome">GDC Reference Genome</a> section below). Each read group is aligned separately. Read group alignments that belong to a single sample are then merged and duplicate reads are flagged for downstream variant calling.</p><p><img src="/warp/assets/images/GDC-f632671b22c77921a8a18af81f65b521.png"></p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="set-up"></a>Set-up<a class="hash-link" href="#set-up" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="workflow-installation-and-requirements"></a>Workflow installation and requirements<a class="hash-link" href="#workflow-installation-and-requirements" title="Direct link to heading">#</a></h3><p>The workflow is written in the Workflow Description Language WDL and can be downloaded by cloning the <a href="https://github.com/broadinstitute/warp/tree/master" target="_blank" rel="noopener noreferrer">warp repository</a> in GitHub. The workflow can be deployed using <a href="https://github.com/broadinstitute/cromwell" target="_blank" rel="noopener noreferrer">Cromwell</a>, a GA4GH compliant, flexible workflow management system that supports multiple computing platforms.</p><p>For the latest workflow version and release notes, please see the <a href="https://github.com/broadinstitute/warp/blob/master/beta-pipelines/broad/somatic/single_sample/wgs/gdc_genome/GDCWholeGenomeSomaticSingleSample.changelog.md" target="_blank" rel="noopener noreferrer">changelog</a>.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="software-version-requirements"></a>Software version requirements<a class="hash-link" href="#software-version-requirements" title="Direct link to heading">#</a></h3><ul><li>GATK 4.0.7</li><li>Picard 2.18.11 (Custom Docker is used to run software on Cromwell 52)</li><li>Samtools 1.11</li><li>Python 3.0</li><li>Cromwell version support<ul><li>Successfully tested on v52</li><li>Does not work on versions &lt; v23 due to output syntax</li></ul></li><li>Papi version support<ul><li>Successfully tested on Papi v2</li></ul></li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="input-descriptions"></a>Input descriptions<a class="hash-link" href="#input-descriptions" title="Direct link to heading">#</a></h3><p>The table below describes each of the GDC pipeline inputs. The workflow requires a <strong>single</strong> aligned CRAM or BAM file, or a <strong>single</strong> unmapped BAM (uBAM) as input, and is set up to run on samples with reads greater than 75 bp.</p><p>For examples of how to specify each input as well as cloud locations for different example input files, see the <a href="https://github.com/broadinstitute/warp/blob/master/beta-pipelines/broad/somatic/single_sample/wgs/gdc_genome/input_files/GDCWholeGenomeSomaticSingleSample.inputs.json" target="_blank" rel="noopener noreferrer">input configuration file (JSON)</a>.</p><table><thead><tr><th>Input name</th><th>Description</th><th>Type</th></tr></thead><tbody><tr><td>input_cram (optional)</td><td>A single mapped CRAM file; alternatively, input can be a mapped BAM (<code>input_bam</code>) or unmapped BAM (uBAM; <code>ubam</code>). If using a CRAM file aligned to a reference different than the reference chosen for the GDC workflow (the <code>ref_fasta</code> input), then you must specify both the <code>cram_ref_fasta</code> and the <code>cram_ref_fasta_index</code> inputs.</td><td>File</td></tr><tr><td>input_bam (optional)</td><td>A single mapped BAM file; alternatively input can be a CRAM (<code>input_cram</code>) or uBAM (<code>ubam</code>).</td><td>File</td></tr><tr><td>cram_ref_fasta (optional)</td><td>The reference file that was used to align the CRAM input (if used); if unspecified, the workflow will use the <code>ref_fasta</code> input file by default.</td><td>File</td></tr><tr><td>cram_ref_fasta_index (optional)</td><td>CRAM reference FASTA index for the <code>cram_ref_fata</code> if CRAM is used as workflow input.</td><td>File</td></tr><tr><td>output_map (optional)</td><td>Tab-separated file containing two columns: a list of all the read group IDs found in the input_cram (or input_bam) and a list of the desired name of the uBAMs generated for read group.</td><td>File</td></tr><tr><td>unmapped_bam_suffix (optional)</td><td>Optional string used to name the output uBAM file.</td><td>String</td></tr><tr><td>ubam (optional)</td><td>A single uBAM file; alternatively, input can be a mapped BAM (<code>input_bam</code>) or CRAM (<code>input_cram</code>).</td><td>File</td></tr><tr><td>contamination_vcf</td><td>VCF file of common variant sites that is used for the check_contamination task.</td><td>File</td></tr><tr><td>contamination_vcf_index</td><td>Index file for the <code>contamination_vcf</code> input.</td><td>File</td></tr><tr><td>dbsnp_vcf</td><td>VCF file of known variation sites that can be used to exclude these sites from the analysis; used for the gatk_baserecalibrator task.</td><td>File</td></tr><tr><td>dbsnp_vcf_index</td><td>Index file for the <code>dbsnp_vcf</code> input.</td><td>File</td></tr><tr><td>ref_fasta</td><td>Reference FASTA used to convert an input CRAM to BAM; if CRAM is not used, this file does not need to be specified.</td><td>File</td></tr><tr><td>ref_fai</td><td>Reference FASTA index for the <code>ref_fasta</code> input.</td><td>File</td></tr><tr><td>ref_dict</td><td>BWA reference dictionary used for alignment.</td><td>File</td></tr><tr><td>ref_amb</td><td>BWA reference file used for alignment.</td><td>File</td></tr><tr><td>ref_ann</td><td>BWA reference file used for alignment.</td><td>File</td></tr><tr><td>ref_bwt</td><td>BWA reference file used for alignment.</td><td>File</td></tr><tr><td>ref_pac</td><td>BWA reference file used for alignment.</td><td>File</td></tr><tr><td>ref_sa</td><td>BWA reference file used for alignment.</td><td>File</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="gdc-reference-genome"></a>GDC reference genome<a class="hash-link" href="#gdc-reference-genome" title="Direct link to heading">#</a></h3><p>The GDC uses the human reference genome GRCh38.d1.vd1 for all data processing. Unlike the GRCh38 reference used by WARP pipelines for production, the GDC reference includes decoy viral sequences for ten types of human viral genomes. You can learn more about the reference from the <a href="https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/#reference-genome" target="_blank" rel="noopener noreferrer">GDC documentation</a>.</p><p>The reference files required for the GDC workflow are hosted in a <a href="https://console.cloud.google.com/storage/browser/gcp-public-data--broad-references/hg38/v0/gdc" target="_blank" rel="noopener noreferrer">public Google Bucket</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="workflow-tasks-and-tools"></a>Workflow tasks and tools<a class="hash-link" href="#workflow-tasks-and-tools" title="Direct link to heading">#</a></h2><p>The <a href="https://github.com/broadinstitute/warp/blob/develop/beta-pipelines/broad/somatic/single_sample/wgs/gdc_genome/GDCWholeGenomeSomaticSingleSample.wdl" target="_blank" rel="noopener noreferrer">workflow</a> imports a series of tasks either from the Workflow script or the Broad <a href="https://github.com/broadinstitute/warp/tree/master/tasks/broad" target="_blank" rel="noopener noreferrer">tasks library</a>.</p><table><thead><tr><th>Task name in WDL</th><th>Tool</th><th>Software</th><th>Description</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/cram_to_unmapped_bams/CramToUnmappedBams.wdl" target="_blank" rel="noopener noreferrer">CramToUnmappedBams</a></td><td>view, index, RevertSam, ValidateSamFile, SortSam</td><td><a href="http://www.htslib.org/terms/" target="_blank" rel="noopener noreferrer">SamTools</a> and <a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Picard</a></td><td>If a CRAM file is used as input, the task converts to uBAM and generates an output map that is then used to split the uBAM by readgroup. The resulting BAM is sorted by query name using Picard.</td></tr><tr><td>bam_readgroup_to_contents</td><td>view</td><td><a href="http://www.htslib.org" target="_blank" rel="noopener noreferrer">Samtools</a></td><td>Extracts all the readgroups from the BAM header and returns a WDL array where each row is a readgroup.</td></tr><tr><td>biobambam_bamtofastq</td><td>bamtofastq</td><td><a href="https://gitlab.com/german.tischler/biobambam2" target="_blank" rel="noopener noreferrer">biobambam2</a></td><td>Converts the uBAMs to FASTQ.</td></tr><tr><td>emit_pe_records/emit_se_records</td><td>---</td><td>---</td><td>Associates the fasta file(s) generated by biobambam_bamtofastq with their respective readgroup; creates an array of structs to be used as input for alignment.</td></tr><tr><td>bwa_pe/ bwa_se</td><td>BWA mem, view</td><td><a href="http://bio-bwa.sourceforge.net/" target="_blank" rel="noopener noreferrer">BWA</a> and <a href="http://www.htslib.org" target="_blank" rel="noopener noreferrer">Samtools</a></td><td>Aligns FASTQ reads to the reference genome, generating an aligned BAM file.</td></tr><tr><td>picard_markduplicates</td><td><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard-" target="_blank" rel="noopener noreferrer">MarkDuplicates</a></td><td><a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Picard</a></td><td>Runs MarkDuplicates in silent mode to locate and tag duplicate reads. Outputs a tagged BAM and metrics file.</td></tr><tr><td>sort_and_index_markdup_bam</td><td>sort</td><td><a href="http://www.htslib.org" target="_blank" rel="noopener noreferrer">Samtools</a></td><td>Sorts and indexes the BAM file; outputs the sorted BAM and index file.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/master/tasks/broad/CheckContaminationSomatic.wdl" target="_blank" rel="noopener noreferrer">check_contamination</a></td><td><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037593451" target="_blank" rel="noopener noreferrer">GetPileUpSummaries</a> and <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036888972" target="_blank" rel="noopener noreferrer">CalculateContamination</a></td><td><a href="https://gatk.broadinstitute.org/hc/en-us" target="_blank" rel="noopener noreferrer">GATK</a></td><td>Summarizes counts of reads that support reference, alternate and other alleles for given sites and then calculates the fraction of reads coming from cross-sample contamination.</td></tr><tr><td>gatk_baserecalibrator</td><td><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360036898312" target="_blank" rel="noopener noreferrer">BaseRecalibrator</a></td><td><a href="https://gatk.broadinstitute.org/hc/en-us" target="_blank" rel="noopener noreferrer">GATK</a></td><td>Generates recalibration table for Base Quality Score Recalibration (BQSR).</td></tr><tr><td>gatk_applybqsr</td><td><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037055712" target="_blank" rel="noopener noreferrer">ApplyBQSR</a></td><td><a href="https://gatk.broadinstitute.org/hc/en-us" target="_blank" rel="noopener noreferrer">GATK</a></td><td>Recalibrates the base qualities of the input reads based on the recalibration table produced by the BaseRecalibrator tool and outputs a recalibrated BAM file; uses the emit_original_quals parameter to write the original base qualities under the BAM OQ tag.</td></tr><tr><td>collect_insert_size_metrics</td><td><a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037055772" target="_blank" rel="noopener noreferrer">CollectInsertSizeMetrics</a></td><td><a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Picard</a></td><td>Generates metrics about insert size distribution in the form of both a histogram and table (txt file).</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="workflow-outputs"></a>Workflow outputs<a class="hash-link" href="#workflow-outputs" title="Direct link to heading">#</a></h2><p>The following table describes the workflow outputs. If running the workflow using Cromwell, these outputs will automatically be placed in the respective task execution directory.</p><p>Alternatively, Cromwell allows you to specify an output directory using an options.json, as described in Cromwell&#x27;s <a href="https://cromwell.readthedocs.io/en/stable/wf_options/Overview/" target="_blank" rel="noopener noreferrer">Workflow Options Overview</a>(see the section on <a href="https://cromwell.readthedocs.io/en/stable/wf_options/Overview/#output-copying" target="_blank" rel="noopener noreferrer">Output Copying</a>).</p><table><thead><tr><th>Output name</th><th>Description</th><th>Type</th></tr></thead><tbody><tr><td>validation_report (optional)</td><td>Samtools validation report(s); only returned if an aligned CRAM or BAM file is used as workflow input</td><td>TXT</td></tr><tr><td>unmapped_bams (optional)</td><td>uBAM file only returned if an aligned CRAM or BAM is used as workflow input.</td><td>BAM</td></tr><tr><td>bam</td><td>Base recalibrated BAM file.</td><td>BAM</td></tr><tr><td>bai</td><td>Index file for the BAM.</td><td>BAI</td></tr><tr><td>md_metrics</td><td>Picard MarkDuplicates metrics.</td><td>TXT</td></tr><tr><td>insert_size_metrics</td><td>Picard insert size metrics.</td><td>TXT</td></tr><tr><td>insert_size_histogram_pdf</td><td>Histogram representation of insert size metrics.</td><td>PDF</td></tr><tr><td>contamination</td><td>File containing a value indicating the fraction contamination.</td><td>TXT</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="important-notes"></a>Important notes<a class="hash-link" href="#important-notes" title="Direct link to heading">#</a></h2><ul><li>Runtime parameters are optimized for Broad&#x27;s Google Cloud Platform implementation.</li><li>Please visit the <a href="https://gatk.broadinstitute.org/hc/en-us/categories/360002310591" target="_blank" rel="noopener noreferrer">GATK Technical Documentation</a> site for further documentation on GATK-related workflows and tools.</li></ul><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="contact-us"></a>Contact us<a class="hash-link" href="#contact-us" title="Direct link to heading">#</a></h2><p>Please help us make our tools better by contacting <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a> for pipeline-related suggestions or questions.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="licensing"></a>Licensing<a class="hash-link" href="#licensing" title="Direct link to heading">#</a></h2><p>Copyright Broad Institute, 2020 | BSD-3</p><p>The workflow script is released under the <strong>WDL open source code license (BSD-3)</strong> (full license text at <a href="https://github.com/broadinstitute/warp/blob/master/LICENSE" target="_blank" rel="noopener noreferrer">https://github.com/broadinstitute/warp/blob/master/LICENSE</a>). However, please note that the programs it calls may be subject to different licenses. Users are responsible for checking that they are authorized to run all programs before running this script.</p><ul><li><a href="https://software.broadinstitute.org/gatk/download/licensing.php" target="_blank" rel="noopener noreferrer">GATK</a></li><li><a href="http://bio-bwa.sourceforge.net/bwa.shtml#13" target="_blank" rel="noopener noreferrer">BWA</a></li><li><a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Picard</a></li><li><a href="http://www.htslib.org/terms/" target="_blank" rel="noopener noreferrer">Samtools</a></li></ul></div><footer class="docusaurus-mt-lg"><div class="row"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Genomic_Data_Commons_Whole_Genome_Somatic/README.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_">Last updated on <b><time datetime="2021-10-06T18:50:18.000Z">10/6/2021</time></b> by <b>Denis Loginov</b></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Exome_Germline_Single_Sample_Pipeline/exome.methods"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« Exome Germline Single Sample v2.4.4 Methods</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Illumina_Genotyping_Arrays_Pipeline/IlluminaGenotypingArray.documentation"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Illumina Genotyping Array Overview Â»</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#introduction-to-the-gdc-whole-genome-somatic-single-sample-pipeline" class="table-of-contents__link">Introduction to the GDC Whole Genome Somatic Single Sample pipeline</a></li><li><a href="#set-up" class="table-of-contents__link">Set-up</a><ul><li><a href="#workflow-installation-and-requirements" class="table-of-contents__link">Workflow installation and requirements</a></li><li><a href="#software-version-requirements" class="table-of-contents__link">Software version requirements</a></li><li><a href="#input-descriptions" class="table-of-contents__link">Input descriptions</a></li><li><a href="#gdc-reference-genome" class="table-of-contents__link">GDC reference genome</a></li></ul></li><li><a href="#workflow-tasks-and-tools" class="table-of-contents__link">Workflow tasks and tools</a></li><li><a href="#workflow-outputs" class="table-of-contents__link">Workflow outputs</a></li><li><a href="#important-notes" class="table-of-contents__link">Important notes</a></li><li><a href="#contact-us" class="table-of-contents__link">Contact us</a></li><li><a href="#licensing" class="table-of-contents__link">Licensing</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright Â© 2021 Copyright Â© Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.90c71525.js"></script>
<script src="/warp/assets/js/main.1fa25a74.js"></script>
</body>
</html>