<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.5">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">scATAC Overview | WARP</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/README"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="scATAC Overview | WARP"><meta data-react-helmet="true" name="description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><meta data-react-helmet="true" property="og:description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/README"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/README" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/README" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.57c5b218.css">
<link rel="preload" href="/warp/assets/js/runtime~main.90c71525.js" as="script">
<link rel="preload" href="/warp/assets/js/main.1fa25a74.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_3Zt9 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">🌜</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">🌞</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button class="clean-btn backToTopButton_35hR" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed menuWithAnnouncementBar_2WvA"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">About WARP</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">WARP Pipelines</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">CEMBA</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Exome Germline Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">GDC Whole Genome Somatic Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Illumina Genotyping Array</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Imputation</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Optimus</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">Single Cell ATAC</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/README">scATAC Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/scatac.methods">scATAC v1.2.0 Methods</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Whole Genome Germline Single Sample</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Contribution Guide</a></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_1CGd"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_3E-R"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="tocCollapsible_1PrD tocMobile_3Hoh"><button type="button" class="clean-btn tocCollapsibleButton_2O1e">On this page</button></div><div class="markdown"><header><h1>scATAC Overview</h1></header><table><thead><tr><th align="center">Pipeline Version</th><th align="center">Date Updated</th><th align="center">Documentation Author</th><th align="center">Questions or Feedback</th></tr></thead><tbody><tr><td align="center"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">scATAC 1.2.0 </a></td><td align="center">January 04 2021</td><td align="center"><a href="mailto:ekiernan@broadinstitute.org" target="_blank" rel="noopener noreferrer">Elizabeth Kiernan</a></td><td align="center">Please file GitHub issues in WARP or contact <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a></td></tr></tbody></table><p><img alt="scATAC_diagram" src="/warp/assets/images/scATAC_diagram-8a049158447e04f48529d186ea9da5f5.png"></p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="introduction"></a>Introduction<a class="hash-link" href="#introduction" title="Direct link to heading">#</a></h2><p>The scATAC Pipeline was developed by the Broad DSP Pipelines team to process single cell/nucleus ATAC-seq datasets. The pipeline is based on the <a href="https://github.com/r3fang/SnapATAC" target="_blank" rel="noopener noreferrer">SnapATAC pipeline</a> described by <a href="https://www.biorxiv.org/content/10.1101/615179v2.full" target="_blank" rel="noopener noreferrer">Fang et al. (2019)</a>. Overall, the pipeline uses the python module <a href="https://github.com/r3fang/SnapTools" target="_blank" rel="noopener noreferrer">SnapTools</a> to align and process paired reads in the form of FASTQ files. It produces an hdf5-structured Snap file that includes a cell-by-bin count matrix. In addition to the Snap file, the final outputs include a GA4GH compliant aligned BAM and QC metrics.</p><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Want to use the scATAC Pipeline for your publication?</h5></div><div class="admonition-content"><p>Check out the <a href="/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/scatac.methods">scATAC Publication Methods</a> to get started!</p></div></div><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="quick-start-table"></a>Quick Start Table<a class="hash-link" href="#quick-start-table" title="Direct link to heading">#</a></h3><table><thead><tr><th>Pipeline Features</th><th>Description</th><th>Source</th></tr></thead><tbody><tr><td>Assay Type</td><td>Single nucleus ATAC-seq</td><td><a href="https://www.biorxiv.org/content/biorxiv/early/2019/05/13/615179.full.pdf" target="_blank" rel="noopener noreferrer">Preprint here </a></td></tr><tr><td>Overall Workflow</td><td>Generates Snap file with cell-by-bin matrix</td><td>Code available from <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/scATAC/scATAC.wdl" target="_blank" rel="noopener noreferrer">GitHub</a></td></tr><tr><td>Workflow Language</td><td>WDL 1.0</td><td><a href="https://github.com/openwdl/wdl" target="_blank" rel="noopener noreferrer">openWDL</a></td></tr><tr><td>Aligner</td><td>BWA</td><td><a href="https://pubmed.ncbi.nlm.nih.gov/19451168/" target="_blank" rel="noopener noreferrer">Li H. and Durbin R., 2009</a></td></tr><tr><td>Data Input File Format</td><td>File format in which sequencing data is provided</td><td>Paired-end FASTQs with cell barcodes appended to read names (read barcode demultiplexing section <a href="https://github.com/r3fang/SnapATAC/wiki/FAQs#whatissnap" target="_blank" rel="noopener noreferrer">here</a>)</td></tr><tr><td>Data Output File Format</td><td>File formats in which scATAC output is provided</td><td><a href="http://samtools.github.io/hts-specs/" target="_blank" rel="noopener noreferrer">BAM</a>, <a href="https://github.com/r3fang/SnapATAC/wiki/FAQs#whatissnap" target="_blank" rel="noopener noreferrer">Snap</a></td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="set-up"></a>Set-up<a class="hash-link" href="#set-up" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="workflow-installation-and-requirements"></a>Workflow Installation and Requirements<a class="hash-link" href="#workflow-installation-and-requirements" title="Direct link to heading">#</a></h3><p>The <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/scATAC/scATAC.wdl" target="_blank" rel="noopener noreferrer">scATAC workflow</a> is written in the Workflow Description Language WDL and can be downloaded by cloning the GitHub <a href="https://github.com/broadinstitute/warp/" target="_blank" rel="noopener noreferrer">WARP repository</a>. The workflow can be deployed using <a href="https://cromwell.readthedocs.io/en/stable/" target="_blank" rel="noopener noreferrer">Cromwell</a>, a GA4GH compliant, flexible workflow management system that supports multiple computing platforms. For the latest workflow version and release notes, please see the scATAC <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/scATAC/scATAC.changelog.md" target="_blank" rel="noopener noreferrer">changelog</a>.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="pipeline-inputs"></a>Pipeline Inputs<a class="hash-link" href="#pipeline-inputs" title="Direct link to heading">#</a></h3><p>The pipeline inputs are detailed in the table below. You can test the workflow by using the <a href="https://github.com/broadinstitute/warp/tree/master/pipelines/skylab/scATAC/example_inputs/human_example.json" target="_blank" rel="noopener noreferrer">human_example.json</a> example configuration file.</p><table><thead><tr><th>Input name</th><th>Input type</th><th>Description</th></tr></thead><tbody><tr><td>input_fastq1</td><td>File</td><td>FASTQ file of the first reads (R1)</td></tr><tr><td>input_fastq2</td><td>File</td><td>FASTQ file of the second reads (R2)</td></tr><tr><td>input_id</td><td>String</td><td>Unique identifier for the sample; will also be used to name the output files</td></tr><tr><td>input_reference</td><td>File</td><td>Reference bundle that is generated with bwa-mk-index-wdl found <a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/accessory_workflows/build_bwa_reference/bwa-mk-index.wdl" target="_blank" rel="noopener noreferrer">here</a></td></tr><tr><td>genome_name</td><td>String</td><td>Name of the genomic reference (name that precedes the “.tar” in the input_reference)</td></tr><tr><td>output_bam</td><td>String</td><td>Name for the output BAM; default is set to the <code>input_id</code> + &quot;_aligned_bam&quot;</td></tr><tr><td>bin_size_list</td><td>String</td><td>List of bin sizes used to generate cell-by-bin matrices; default is 10000 bp</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="input-file-preparation"></a>Input File Preparation<a class="hash-link" href="#input-file-preparation" title="Direct link to heading">#</a></h3><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="r1-and-r2-fastq-preparation"></a>R1 and R2 FASTQ Preparation<a class="hash-link" href="#r1-and-r2-fastq-preparation" title="Direct link to heading">#</a></h4><p>The scATAC workflow requires paired reads in the form FASTQ files with the cell barcodes appended to the readnames. A description of the barcode demultiplexing can be found on the SnapATAC documentation (see barcode demultiplexing section <a href="https://github.com/r3fang/SnapATAC/wiki/FAQs#CEMBA_snap" target="_blank" rel="noopener noreferrer">here</a>). The full cell barcode must form the first part of the read name (for both R1 and R2 files) and be separated from the rest of the line by a colon. You can find an example python code to perform demultiplexing in the <a href="https://github.com/r3fang/SnapTools/blob/master/snaptools/dex_fastq.py" target="_blank" rel="noopener noreferrer">SnapTools documentation here</a>. The codeblock below demonstrates the correct format.</p><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly"><pre tabindex="0" class="prism-code language-undefined codeBlock_23N8 thin-scrollbar" style="color:#bfc7d5;background-color:#292d3e"><code class="codeBlockLines_39YC"><span class="token-line" style="color:#bfc7d5"><span class="token plain">@CAGTTGCACGTATAGAACAAGGATAGGATAAC:7001113:915:HJ535BCX2:1:1106:1139:1926 1:N:0:0</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">ACCCTCCGTGTGCCAGGAGATACCATGAATATGCCATAGAACCTGTCTCT</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">+</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">DDDDDIIIIIIIIIIIIIIHHIIIIIIIIIIIIIIIIIIIIIIIIIIIII</span></span></code></pre><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o clean-btn">Copy</button></div></div><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="input_reference-preparation"></a>Input_reference Preparation<a class="hash-link" href="#input_reference-preparation" title="Direct link to heading">#</a></h4><p>The input_reference is a BWA compatible reference bundle in TAR file format. You can create this BWA reference using the accessory workflow  <a href="https://github.com/broadinstitute/warp/tree/develop/tasks/skylab/accessory_workflows/build_bwa_reference/bwa-mk-index.wdl" target="_blank" rel="noopener noreferrer">here</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="workflow-tasks-and-tools"></a>Workflow Tasks and Tools<a class="hash-link" href="#workflow-tasks-and-tools" title="Direct link to heading">#</a></h2><p>The <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/scATAC/scATAC.wdl" target="_blank" rel="noopener noreferrer">scATAC workflow</a> is divided into multiple tasks which are described in the table below. The table also links to the Docker Image for each task and to the documentation or code for the relevant software tool parameters.</p><table><thead><tr><th>Task</th><th>Task Description</th><th>Tool Docker Image</th><th>Parameter Descriptions or Code</th></tr></thead><tbody><tr><td>AlignPairedEnd</td><td>Align the modified FASTQ files to the genome</td><td><a href="https://github.com/broadinstitute/warp/blob/master/dockers/skylab/snaptools/Dockerfile" target="_blank" rel="noopener noreferrer">snaptools:0.0.1</a></td><td><a href="https://github.com/r3fang/SnapTools" target="_blank" rel="noopener noreferrer">SnapTools documentation</a></td></tr><tr><td>SnapPre</td><td>Initial generation of snap file</td><td><a href="https://github.com/broadinstitute/warp/blob/master/dockers/skylab/snaptools/Dockerfile" target="_blank" rel="noopener noreferrer">snaptools:0.0.1</a></td><td><a href="https://github.com/r3fang/SnapTools" target="_blank" rel="noopener noreferrer">SnapTools documentation</a></td></tr><tr><td>SnapCellByBin</td><td>Binning of data by genomic bins</td><td><a href="https://github.com/broadinstitute/warp/blob/master/dockers/skylab/snaptools/Dockerfile" target="_blank" rel="noopener noreferrer">snaptools:0.0.1</a></td><td><a href="https://github.com/r3fang/SnapTools" target="_blank" rel="noopener noreferrer">SnapTools documentation</a></td></tr><tr><td>MakeCompliantBAM</td><td>Generation of a GA4GH compliant BAM</td><td><a href="https://github.com/broadinstitute/warp/blob/master/dockers/skylab/snaptools/Dockerfile" target="_blank" rel="noopener noreferrer">snaptools:0.0.1</a></td><td><a href="https://github.com/broadinstitute/warp/blob/master/dockers/skylab/snaptools/makeCompliantBAM.py" target="_blank" rel="noopener noreferrer">Code</a></td></tr><tr><td>BreakoutSnap</td><td>Extraction of tables from snap file into text format (for testing and user availability)</td><td><a href="https://github.com/broadinstitute/warp/tree/master/dockers/skylab/snap-breakout" target="_blank" rel="noopener noreferrer">snap-breakout:0.0.1</a></td><td><a href="https://github.com/broadinstitute/warp/tree/master/dockers/skylab/snap-breakout/breakoutSnap.py" target="_blank" rel="noopener noreferrer">Code</a></td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="task-summary"></a>Task Summary<a class="hash-link" href="#task-summary" title="Direct link to heading">#</a></h3><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="alignpairedend"></a>AlignPairedEnd<a class="hash-link" href="#alignpairedend" title="Direct link to heading">#</a></h4><p>The AlignPairedEnd task takes the barcode demultiplexed FASTQ files and aligns reads to the genome using the BWA aligner. It uses the SnapTools min_cov parameter to set the minimum number of barcodes a fragment requires to be included in the final output. This parameter is set to 0. The final task output is an aligned BAM file.</p><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="snappre"></a>SnapPre<a class="hash-link" href="#snappre" title="Direct link to heading">#</a></h4><p>The SnapPre task uses SnapTools to perform preprocessing and filtering on the aligned BAM. The task outputs are a Snap file and QC metrics. The table below details the filtering parameters for the task.</p><h5><a aria-hidden="true" tabindex="-1" class="anchor anchor__h5 anchorWithStickyNavbar_31ik" id="filtering-parameters"></a>Filtering Parameters<a class="hash-link" href="#filtering-parameters" title="Direct link to heading">#</a></h5><table><thead><tr><th>Parameter</th><th>Description</th><th>Value</th></tr></thead><tbody><tr><td>--min-mapq</td><td>Fragments with mappability less than value will be filtered</td><td>30</td></tr><tr><td>--min-flen</td><td>Fragments of length shorter than min_flen will be filtered</td><td>0</td></tr><tr><td>--max-flen</td><td>Fragments of length bigger than min_flen will be filtered</td><td>1000</td></tr><tr><td>--keep-chrm</td><td>Boolean variable indicates whether to keep reads mapped to chrM</td><td>TRUE</td></tr><tr><td>--keep-single</td><td>Boolean variable indicates whether to keep single-end reads</td><td>TRUE</td></tr><tr><td>--keep-secondary</td><td>Boolean variable indicates whether to keep secondary alignments</td><td>FALSE</td></tr><tr><td>--max-num</td><td>Max number of barcodes to be stored. Based on the coverage, top max_barcode barcodes are selected and stored</td><td>1000000</td></tr><tr><td>--min-cov</td><td>Fragments with less than min-cov number of barcodes will be filtered</td><td>100</td></tr></tbody></table><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="snapcellbybin"></a>SnapCellByBin<a class="hash-link" href="#snapcellbybin" title="Direct link to heading">#</a></h4><p>The SnapCellByBin task uses the Snap file to create cell-by-bin count matrices in which a “1” represents a bin with an accessible region of the genome and a “0” represents an inaccessible region. The bin_size_list sets the bin size to 10,000 bp by default but can be changed by specifying the value in the inputs to the workflow.</p><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="makecompliantbam"></a>MakeCompliantBAM<a class="hash-link" href="#makecompliantbam" title="Direct link to heading">#</a></h4><p>The MakeCompliantBAM task uses a <a href="https://github.com/broadinstitute/warp/blob/master/dockers/skylab/snaptools/makeCompliantBAM.py" target="_blank" rel="noopener noreferrer">custom python script (here)</a> to make a GA4GH compliant BAM by moving the cellular barcodes in the read names to the CB tag.</p><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="breakoutsnap"></a>BreakoutSnap<a class="hash-link" href="#breakoutsnap" title="Direct link to heading">#</a></h4><p>The BreakoutSnap task extracts data from the Snap file and exports it to individual CSVs. These CSV outputs are listed in the table in the Outputs section below. The code is available <a href="https://github.com/broadinstitute/warp/tree/master/dockers/skylab/snap-breakout/breakoutSnap.py" target="_blank" rel="noopener noreferrer">here</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="outputs"></a>Outputs<a class="hash-link" href="#outputs" title="Direct link to heading">#</a></h2><p>The main outputs of the scATAC workflow are the Snap file, Snap QC metrics, and the GA4GH compliant BAM file. All files with the prefix “breakout” are CSV files containing individual pieces of data from the Snap. The sessions for the Snap file are described in the <a href="https://github.com/r3fang/SnapTools" target="_blank" rel="noopener noreferrer">SnapTools documentation</a>. Additionally, you can read detailed information on the <a href="https://github.com/r3fang/SnapTools/blob/master/docs/snap_format.docx" target="_blank" rel="noopener noreferrer">Snap file fields for each session</a> (select &quot;View Raw&quot;).</p><table><thead><tr><th>Output File Name</th><th>Description</th></tr></thead><tbody><tr><td>output_snap_qc</td><td>Quality control file corresponding to the snap file</td></tr><tr><td>output_snap</td><td>Output snap file (in hdf5 container format)</td></tr><tr><td>output_aligned_bam</td><td>Output BAM file, compliant with GA4GH standards</td></tr><tr><td>breakout_barcodes</td><td>Text file containing the FM (&#x27;Fragment session&#x27;) barcodeLen and barcodePos fields</td></tr><tr><td>breakout_fragments</td><td>Text file containing the FM (&#x27;Fragments session&#x27;) fragChrom, fragLen, and fragStart fields</td></tr><tr><td>breakout_binCoordinates</td><td>Text file with the AM session (&#x27;Cell x bin accessibility&#x27; matrix) binChrom and binStart fields</td></tr><tr><td>breakout_binCounts</td><td>Text file with the AM session (&#x27;Cell x bin accessibility&#x27; matrix) idx, idy, and count fields</td></tr><tr><td>breakout_barcodesSection</td><td>Text file with the data from the BD session (&#x27;Barcode session&#x27; table)</td></tr></tbody></table><h5><a aria-hidden="true" tabindex="-1" class="anchor anchor__h5 anchorWithStickyNavbar_31ik" id="snap-qc-metrics"></a>Snap QC Metrics<a class="hash-link" href="#snap-qc-metrics" title="Direct link to heading">#</a></h5><p>The following table details the metrics available in the output_snap_qc file.</p><table><thead><tr><th>QC Metric</th><th>Abbreviation</th></tr></thead><tbody><tr><td>Total number of unique barcodes</td><td>No abbreviation</td></tr><tr><td>Total number of fragments</td><td>TN</td></tr><tr><td>Total number of uniquely mapped</td><td>UM</td></tr><tr><td>Total number of single ends</td><td>SE</td></tr><tr><td>Total number of secondary alignments</td><td>SA</td></tr><tr><td>Total number of paired ends</td><td>PE</td></tr><tr><td>Total number of proper paired</td><td>PP</td></tr><tr><td>Total number of proper frag len</td><td>PL</td></tr><tr><td>Total number of usable fragments</td><td>US</td></tr><tr><td>Total number of unique fragments</td><td>UQ</td></tr><tr><td>Total number of chrM fragments</td><td>CM</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="running-on-terra"></a>Running on Terra<a class="hash-link" href="#running-on-terra" title="Direct link to heading">#</a></h2><p><a href="https://app.terra.bio/" target="_blank" rel="noopener noreferrer">Terra</a> is a public, cloud-based platform for biomedical research. If you would like to try the scATAC workflow (previously named &quot;snap-atac&quot;) in Terra, you can import the most recent version from <a href="https://dockstore.org/workflows/github.com/broadinstitute/warp/scATAC:scATAC_v1.1.0?tab=info" target="_blank" rel="noopener noreferrer">Dockstore</a>. Additionally, there is a public <a href="https://app.terra.bio/#workspaces/brain-initiative-bcdc/scATAC" target="_blank" rel="noopener noreferrer">scATAC workspace</a> preloaded with the scATAC workflow and downsampled data.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="versioning"></a>Versioning<a class="hash-link" href="#versioning" title="Direct link to heading">#</a></h2><p>All scATAC workflow releases are documented in the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/scATAC/scATAC.changelog.md" target="_blank" rel="noopener noreferrer">scATAC changelog</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="citing-the-scatac-pipeline"></a>Citing the scATAC Pipeline<a class="hash-link" href="#citing-the-scatac-pipeline" title="Direct link to heading">#</a></h2><p>Please identify the pipeline in your methods section using the scATAC Pipeline&#x27;s <a href="https://scicrunch.org/scicrunch/Resources/record/nlx_144509-1/SCR_018919/resolver?q=SCR_018919&amp;l=SCR_018919" target="_blank" rel="noopener noreferrer">SciCrunch resource identifier</a>.</p><ul><li>Ex: <em>scATAC Pipeline (RRID:SCR_018919)</em></li></ul><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="consortia-support"></a>Consortia Support<a class="hash-link" href="#consortia-support" title="Direct link to heading">#</a></h2><p>This pipeline is supported and used by the <a href="https://biccn.org/" target="_blank" rel="noopener noreferrer">BRAIN Initiative Cell Census Network</a> (BICCN). </p><p>If your organization also uses this pipeline, we would love to list you! Please reach out to us by contacting <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="pipeline-improvements"></a>Pipeline Improvements<a class="hash-link" href="#pipeline-improvements" title="Direct link to heading">#</a></h2><p>Please help us make our tools better by contacting <a href="mailto:kdegatano@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kylee Degatano</a> for pipeline-related suggestions or questions.</p></div><footer class="docusaurus-mt-lg"><div class="row"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/README.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_">Last updated on <b><time datetime="2021-10-06T18:50:18.000Z">10/6/2021</time></b> by <b>Denis Loginov</b></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Optimus_Pipeline/consortia-processing"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">« Consortia Data Processing</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Single_Cell_ATAC_Seq_Pipeline/scatac.methods"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">scATAC v1.2.0 Methods »</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#introduction" class="table-of-contents__link">Introduction</a><ul><li><a href="#quick-start-table" class="table-of-contents__link">Quick Start Table</a></li></ul></li><li><a href="#set-up" class="table-of-contents__link">Set-up</a><ul><li><a href="#workflow-installation-and-requirements" class="table-of-contents__link">Workflow Installation and Requirements</a></li><li><a href="#pipeline-inputs" class="table-of-contents__link">Pipeline Inputs</a></li><li><a href="#input-file-preparation" class="table-of-contents__link">Input File Preparation</a></li></ul></li><li><a href="#workflow-tasks-and-tools" class="table-of-contents__link">Workflow Tasks and Tools</a><ul><li><a href="#task-summary" class="table-of-contents__link">Task Summary</a></li></ul></li><li><a href="#outputs" class="table-of-contents__link">Outputs</a></li><li><a href="#running-on-terra" class="table-of-contents__link">Running on Terra</a></li><li><a href="#versioning" class="table-of-contents__link">Versioning</a></li><li><a href="#citing-the-scatac-pipeline" class="table-of-contents__link">Citing the scATAC Pipeline</a></li><li><a href="#consortia-support" class="table-of-contents__link">Consortia Support</a></li><li><a href="#pipeline-improvements" class="table-of-contents__link">Pipeline Improvements</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2021 Copyright © Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.90c71525.js"></script>
<script src="/warp/assets/js/main.1fa25a74.js"></script>
</body>
</html>