<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.5">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">Ultima Genomics Whole Genome Germline Overview | WARP</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/Ultima_Genomics_Whole_Genome_Germline_Pipeline/README"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Ultima Genomics Whole Genome Germline Overview | WARP"><meta data-react-helmet="true" name="description" content="| Pipeline Version | Date Updated | Documentation Authors | Questions or Feedback |"><meta data-react-helmet="true" property="og:description" content="| Pipeline Version | Date Updated | Documentation Authors | Questions or Feedback |"><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/Ultima_Genomics_Whole_Genome_Germline_Pipeline/README"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Ultima_Genomics_Whole_Genome_Germline_Pipeline/README" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Ultima_Genomics_Whole_Genome_Germline_Pipeline/README" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.57c5b218.css">
<link rel="preload" href="/warp/assets/js/runtime~main.e360ee0b.js" as="script">
<link rel="preload" href="/warp/assets/js/main.143bdc00.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_3Zt9 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">ðŸŒž</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button class="clean-btn backToTopButton_35hR" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed menuWithAnnouncementBar_2WvA"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">About WARP</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">WARP Pipelines</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">CEMBA</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Exome Germline Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">GDC Whole Genome Somatic Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Illumina Genotyping Array</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Imputation</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Optimus</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">RNA with UMIs</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Single Cell ATAC</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Sample</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">Ultima Genomics Whole Genome Germline</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/warp/docs/Pipelines/Ultima_Genomics_Whole_Genome_Germline_Pipeline/README">Ultima Genomics Whole Genome Germline Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Whole Genome Germline Single Sample</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Contribution Guide</a></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_1CGd"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_3E-R"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="tocCollapsible_1PrD tocMobile_3Hoh"><button type="button" class="clean-btn tocCollapsibleButton_2O1e">On this page</button></div><div class="markdown"><header><h1>Ultima Genomics Whole Genome Germline Overview</h1></header><table><thead><tr><th align="center">Pipeline Version</th><th align="center">Date Updated</th><th align="center">Documentation Authors</th><th align="center">Questions or Feedback</th></tr></thead><tbody><tr><td align="center"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">UltimaGenomicsWholeGenomeGermline_v1.0.0</a></td><td align="center">June, 2022</td><td align="center"><a href="mailto:ekiernan@broadinstitute.org" target="_blank" rel="noopener noreferrer">Elizabeth Kiernan</a> &amp; <a href="mailto:kmathews@broadinstitute.org" target="_blank" rel="noopener noreferrer">Kaylee Mathews</a></td><td align="center">Please file GitHub issues in warp or contact <a href="mailto:warp-pipelines-help@broadinstitute.org" target="_blank" rel="noopener noreferrer">the wARP team</a></td></tr></tbody></table><p><img alt="UG_diagram" src="/warp/assets/images/ug_diagram-f967ed2cc6a58e654c0a27f1e9bf8d3b.png"></p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="introduction-to-the-ug_wgs-workflow"></a>Introduction to the UG_WGS workflow<a class="hash-link" href="#introduction-to-the-ug_wgs-workflow" title="Direct link to heading">#</a></h2><p>The <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/germline/single_sample/ugwgs/UltimaGenomicsWholeGenomeGermline.wdl" target="_blank" rel="noopener noreferrer">Ultima Genomics Whole Genome Germline (UG_WGS) workflow</a> is an open-source, cloud-optimized workflow for processing whole-genome germline sequencing data generated using the Ultima Genomics sequencing platform. </p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="background-ultima-genomics-sequencing"></a>Background: Ultima Genomics sequencing<a class="hash-link" href="#background-ultima-genomics-sequencing" title="Direct link to heading">#</a></h3><p>Ultima Genomics sequencing is a novel technology that produces single-read, flow-based data (<a href="https://www.biorxiv.org/content/10.1101/2022.05.29.493900v1" target="_blank" rel="noopener noreferrer">Almogy et al., 2022</a>). The sequencing platform works by flowing one nucleotide at a time in order, iteratively. This is in contrast to traditional technologies that do all four nucleotides at once. This iterative approach ensures that only one dNTP is responsible for the signal and it does not require the blocking of dNTPs.  </p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="what-does-the-workflow-do"></a>What does the workflow do?<a class="hash-link" href="#what-does-the-workflow-do" title="Direct link to heading">#</a></h3><p>The workflow requires either an aligned CRAM output of the sequencing platform or an unmapped BAM as input. Overall, it aligns reads to a reference genome, marks duplicate reads, calls variants, post-processes variants in the output VCF in preparation for joint calling, and calculates quality control metrics. The workflow outputs a (re)aligned CRAM, an annotated GVCF with index, and quality metrics. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="set-up"></a>Set-up<a class="hash-link" href="#set-up" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="installation"></a>Installation<a class="hash-link" href="#installation" title="Direct link to heading">#</a></h3><p>To download the latest release of the UG_WGS pipeline, see the release tags prefixed with &quot;UG_WGS&quot; on the WARP <a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">releases page</a>. All releases of the UG_WGS pipeline are documented in the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/germline/single_sample/ugwgs/UltimaGenomicsWholeGenomeGermline.changelog.md" target="_blank" rel="noopener noreferrer">UG_WGS changelog</a>. </p><p>To search releases of this and other pipelines, use the WARP command-line tool <a href="https://github.com/broadinstitute/warp/tree/develop/wreleaser" target="_blank" rel="noopener noreferrer">Wreleaser</a>.</p><p>The UG_WGS pipeline can be deployed using <a href="https://cromwell.readthedocs.io/en/stable/" target="_blank" rel="noopener noreferrer">Cromwell</a>, a GA4GH compliant, flexible workflow management system that supports multiple computing platforms. The workflow can also be run in <a href="https://app.terra.bio" target="_blank" rel="noopener noreferrer">Terra</a>, a cloud-based analysis platform. </p><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="inputs"></a>Inputs<a class="hash-link" href="#inputs" title="Direct link to heading">#</a></h3><p>The UG_WGS workflow inputs are specified in JSON configuration files. An example configuration file can be found in the <a href="https://github.com/broadinstitute/warp/tree/develop/pipelines/broad/dna_seq/germline/single_sample/ugwgs/input_files" target="_blank" rel="noopener noreferrer">input_files</a> folder in the WARP repository. </p><p>Multiple workflow inputs are in the form of structs, which are defined in <a href="https://github.com/broadinstitute/warp/blob/develop/structs/dna_seq/UltimaGenomicsWholeGenomeGermlineStructs.wdl" target="_blank" rel="noopener noreferrer">UG_WGS structs WDL</a>. All structs are listed in the UltimaGenomicsWholeGenomeGermlineStructs WDL](<a href="https://github.com/broadinstitute/warp/develop/structs/dna_seq/UltimaGenomicsWholeGenomeGermlineStructs.wdl" target="_blank" rel="noopener noreferrer">https://github.com/broadinstitute/warp/develop/structs/dna_seq/UltimaGenomicsWholeGenomeGermlineStructs.wdl</a>).</p><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="input-descriptions"></a>Input descriptions<a class="hash-link" href="#input-descriptions" title="Direct link to heading">#</a></h4><p>The workflow takes in an aligned CRAM (output of the Ultima Genomics sequencing platform) or an unmapped BAM file for one sample and one read group. </p><p>The workflow input variables are listed below. If an input variable is part of a struct, the struct name is listed in the <code>Struct</code> column.</p><table><thead><tr><th>Input variable name</th><th>Struct</th><th>Description</th><th>Type</th></tr></thead><tbody><tr><td>input_cram_list</td><td>N/A</td><td>Array of CRAM files to be used as workflow input; must be specified if <code>input_bam_list</code> is not provided.</td><td>Array [File]</td></tr><tr><td>input_bam_list</td><td>N/A</td><td>Array of unmapped BAM files to be used as workflow input; must be specified if <code>input_cram_list</code> is not provided.</td><td>Array [File]</td></tr><tr><td>base_file_name</td><td>N/A</td><td>Base name for each of the output files.</td><td>String</td></tr><tr><td>contamination_sites_path</td><td>ContaminationSites</td><td>Path to contamination site files.</td><td>String</td></tr><tr><td>contamination_sites_vcf</td><td>ContaminationSites</td><td>Contamination site VCF.</td><td>File</td></tr><tr><td>contamination_sites_vcf_index</td><td>ContaminationSites</td><td>Index for contamination site VCF.</td><td>File</td></tr><tr><td>ref_fasta</td><td>References</td><td>Reference FASTA file used for alignment with BWA-MEM.</td><td>File</td></tr><tr><td>ref_fasta_index</td><td>References</td><td>Reference FASTA index file used for alignment with BWA-MEM.</td><td>File</td></tr><tr><td>ref_dict</td><td>References</td><td>Dictionary file used for alignment with BWA-MEM.</td><td>File</td></tr><tr><td>ref_alt</td><td>AlignmentReferences</td><td>Reference files used for alignment with BWA-MEM.</td><td>File</td></tr><tr><td>ref_amb</td><td>AlignmentReferences</td><td>Reference files used for alignment with BWA-MEM.</td><td>File</td></tr><tr><td>ref_ann</td><td>AlignmentReferences</td><td>Reference files used for alignment with BWA-MEM.</td><td>File</td></tr><tr><td>ref_bwt</td><td>AlignmentReferences</td><td>Reference files used for alignment with BWA-MEM.</td><td>File</td></tr><tr><td>ref_pac</td><td>AlignmentReferences</td><td>Reference files used for alignment with BWA-MEM.</td><td>File</td></tr><tr><td>ref_sa</td><td>AlignmentReferences</td><td>Reference files used for alignment with BWA-MEM.</td><td>File</td></tr><tr><td>wgs_calling_interval_list</td><td>VariantCallingSettings</td><td>Interval list used for variant calling with HaplotypeCaller.</td><td>File</td></tr><tr><td>break_bands_at_multiples_of</td><td>VariantCallingSettings</td><td>Breaks reference bands up at genomic positions that are multiples of this number; used to reduce GVCF file size.</td><td>Int</td></tr><tr><td>haplotype_scatter_count</td><td>VariantCallingSettings</td><td>Scatter count used for variant calling.</td><td>Int</td></tr><tr><td>make_haplotype_bam</td><td>N/A</td><td>Boolean indicating if HaplotypeCaller should output a bamout file; default is set to &quot;false&quot;.</td><td>Boolean</td></tr><tr><td>rsq_threshold</td><td>NA</td><td>Threshold for a read quality metric that is produced by the sequencing platform; default set to 1.0</td><td>Float</td></tr><tr><td>annotation_intervals</td><td>VcfPostProcessing</td><td>Annotation intervals used for filtering and annotating the HaplotypeCaller output VCF.</td><td>Array[File]</td></tr><tr><td>filtering_model_no_gt</td><td>VcfPostProcessing</td><td>Optional filtering file defining the model for VCF postprocessing.</td><td>File</td></tr><tr><td>af_only_gnomad</td><td>VcfPostProcessing</td><td>VCF with gnomAD allele frequencies used for metric stratification in the AnnotateVCF_AF task.</td><td>File</td></tr><tr><td>af_only_gnomad_index</td><td>VcfPostProcessing</td><td>Index for the <code>af_only_gnomad</code> VCF.</td><td>File</td></tr><tr><td>filter_cg_insertions</td><td>VcfPostProcessing</td><td>Boolean that indicates whether to filter CG insertions in HaplotypeCaller output VCF.</td><td>Boolean</td></tr><tr><td>filtering_blocklist_file</td><td>VcfPostProcessing</td><td>Optional file used to flag genomic locations that canâ€™t be trusted while filtering the HaplotypeCaller output VCF.</td><td>File</td></tr><tr><td>training_blocklist_file</td><td>VcfPostProcessing</td><td>Optional interval file for training a model to postprocess the HaplotypeCaller output VCF to identify false positives.</td><td>File</td></tr><tr><td>exome_weight</td><td>VcfPostProcessing</td><td>Optional interval for exome weight that is used to train a model to identify false positives in the HaplotypeCaller output VCF.</td><td>Int</td></tr><tr><td>exome_weight_annotation</td><td>VcfPostProcessing</td><td>Optional string to annotate exome weights.</td><td>String</td></tr><tr><td>interval_list_override</td><td>VcfPostProcessing</td><td>Optional interval list to use for VCF postprocessing</td><td>File</td></tr><tr><td>runs_file</td><td>VcfPostProcessing</td><td>BED file of homopolymer runs that can be difficult to sequence.</td><td>File</td></tr><tr><td>filtering_model_with_gt_name_override</td><td>VcfPostProcessing</td><td>Optional string to describe the optional filtering model with gt.</td><td>String</td></tr><tr><td>max_duplication_in_reasonable_sample</td><td>VcfPostProcessing</td><td>Number indicating the maximum duplication expected in a sample.</td><td>Float</td></tr><tr><td>max_chimerism_in_reasonable_sample</td><td>VcfPostProcessing</td><td>Number indicating the maximum chimerism expected in a sample.</td><td>Float</td></tr><tr><td>ref_dbsnp</td><td>VcfPostProcessing</td><td>VCF with dbSNP annotations to be added to the HaplotypeCaller output VCF.</td><td>File</td></tr><tr><td>ref_dbsnp_index</td><td>VcfPostProcessing</td><td>Index for the dbSNP annotation VCF.</td><td>File</td></tr><tr><td>wgs_coverage_interval_list</td><td>VcfPostProcessing</td><td>Interval list for the CollectWgsMetrics tool.</td><td>File</td></tr><tr><td>remove_low_tree_score_sites_cutoff</td><td>VcfPostProcessing</td><td>Optional float indicating a cutoff for low tree scores</td><td>Float</td></tr><tr><td>annotations_to_keep_command_for_reblocking</td><td>VcfPostProcessing</td><td>Optional command (--annotations-to-keep) for the ReblockGVCF task indicating what types annotations to keep in the reblocked GVCF.</td><td>String</td></tr><tr><td>increase_disk_size</td><td>N/A</td><td>Interval used to increase disk size; set to 20 GB by default.</td><td>Int</td></tr><tr><td>increase_metrics_disk_size</td><td>N/A</td><td>Interval used to adjust disk size; set to 80 by default.</td><td>Int</td></tr><tr><td>filtering_model_no_gt_name</td><td>N/A</td><td>String to describe the optional filtering model; default set to &quot;rf_model_ignore_gt_incl_hpol_runs&quot;.</td><td>String</td></tr><tr><td>merge_bam_file</td><td>N/A</td><td>Boolean indicating if by-interval bamout files from HaplotypeCaller should be merged into a single BAM.</td><td>Boolean</td></tr><tr><td>reads_per_split</td><td>N/A</td><td>Number of reads by which to split the CRAM prior to alignment.</td><td>Int</td></tr><tr><td>pipeline_version</td><td>N/A</td><td>String that describes the pipeline version number.</td><td>String</td></tr></tbody></table><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="reference-files"></a>Reference files<a class="hash-link" href="#reference-files" title="Direct link to heading">#</a></h4><p>Reference files, such as the hg38- and dbSNP-related files are located in a public Google bucket. See the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/germline/single_sample/ugwgs/input_files/UltimaGenomicsWholeGenomeGermline.inputs.json" target="_blank" rel="noopener noreferrer">example input configuration file</a> for cloud locations.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="ug_wgs-tasks-and-tools"></a>UG_WGS tasks and tools<a class="hash-link" href="#ug_wgs-tasks-and-tools" title="Direct link to heading">#</a></h2><p>The <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/germline/single_sample/ugwgs/UltimaGenomicsWholeGenomeGermline.wdl" target="_blank" rel="noopener noreferrer">UG_WGS workflow</a> imports additional WDL scripts that contain the different workflow tasks. When applicable, links to these additional WDL scripts (subtasks/subworkflows) are provided in the summary section below. </p><ul><li><p>Workflow tasks use different software tools to manipulate the workflow input data. To see specific tool parameters, select the task WDL link in the table; then find the task and view the <code>command {}</code> section of the task in the WDL script. </p></li><li><p>To view or use the exact tool software, see the task&#x27;s Docker image which is specified in the task WDL <code>input {}</code> section as <code>String docker =</code>. </p></li><li><p>Docker images for the UG_WGS workflow are not yet versioned and officially released, but they are publicly available to test the workflow. </p></li></ul><p>Overall, the UG_WGS workflow:</p><ol><li>Aligns with BWA-MEM and marks duplicates.</li><li>Converts a merged BAM to CRAM and validates the CRAM.</li><li>Extracts the nucleotide flow order.</li><li>Calculates quality control metrics.</li><li>Performs variant calling with HaplotypeCaller.</li><li>Merges BAMs and converts the GVCF to VCF.</li><li>Performs VCF post-processing in preparation for joint calling.</li></ol><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>Multiple tasks are imported from nested sub-workflows</h5></div><div class="admonition-content"><p>When applicable, links to the sub-workflow WDLs and nested tasks WDLs are provided. </p></div></div><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="1-align-and-mark-duplicates"></a>1. Align and mark duplicates<a class="hash-link" href="#1-align-and-mark-duplicates" title="Direct link to heading">#</a></h3><p><strong>Sub-workflow name and link:</strong> <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/somatic/single_sample/ugwgs/UltimaGenomicsWholeGenomeCramOnly.wdl" target="_blank" rel="noopener noreferrer">UltimaGenomicsWholeGenomeCramOnly</a></p><ul><li><p><strong>Nested task name and WDL link:</strong></p><p><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineAlignmentMarkDuplicates.wdl" target="_blank" rel="noopener noreferrer">UltimaGenomicsWholeGenomeGermlineAlignmentMarkDuplicates.AlignmentAndMarkDuplicates</a></p></li></ul><p>The table below details the subtasks called by the AlignmentAndMarkDuplicates task, which splits the CRAM or BAM into subfiles, converts them to uBAM format, converts the uBAM to FASTQ for alignment with BWA-MEM, and marks duplicate reads in the resulting BAM files. </p><p>The Picard tool MarkDuplicatesSpark has been adapted to handle ambiguity in the aligned start of a read. This functionality is applied using the toolâ€™s <code>--flowbased</code> parameter.</p><table><thead><tr><th>Subtask name and WDL link</th><th>Tool</th><th>Software</th><th>Description</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.SplitCram as SplitInputCram</a></td><td>split</td><td>crammer</td><td>If CRAM is the workflow input, splits the CRAM and outputs a split CRAM.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/Alignment.wdl" target="_blank" rel="noopener noreferrer">AlignmentTasks.SamSplitter as SplitInputBam</a></td><td>SplitSamByNumberOfReads</td><td>Picard</td><td>If BAM is workflow input, splits the BAM and outputs an array of BAMs.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.ConvertCramOrBamToUBam as ConvertToUbam</a></td><td>view, RevertSam</td><td>samtools, Picard</td><td>Converts the split CRAM or BAM file to uBAM.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.SamToFastqAndBwaMemAndMba</a></td><td>SamToFastq, bwa mem, MergeBamAlignment</td><td>Picard, bwa mem</td><td>Converts each uBAM to FASTQ format, aligns with BWA-MEM, and merges the alignment in the resulting BAM with metadata from the uBAM.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.MarkDuplicatesSpark</a></td><td>MarkDuplicatesSpark</td><td>GATK</td><td>Flags duplicate reads in the array of aligned and merged BAMs to create a new output BAM and index.</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="2-convert-bam-to-cram-and-validate-the-cram"></a>2. Convert BAM to CRAM and validate the CRAM<a class="hash-link" href="#2-convert-bam-to-cram-and-validate-the-cram" title="Direct link to heading">#</a></h3><p><strong>Sub-workflow name and link:</strong> <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/UltimaGenomics/UltimaGenomicsWholeGenomeCramOnly.wdl" target="_blank" rel="noopener noreferrer">UltimaGenomicsWholeGenomeCramOnly</a></p><p>The two tasks below are used to convert each duplicate-marked BAM to CRAM and validate the resulting files.</p><table><thead><tr><th>TASK name and WDL link</th><th>Tool</th><th>Software</th><th>Description</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/Utilities.wdl" target="_blank" rel="noopener noreferrer">Utilities.ConvertToCram</a></td><td>view</td><td>samtools</td><td>Converts the duplicated-marked BAM to CRAM.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/Qc.wdl" target="_blank" rel="noopener noreferrer">QC.ValidateSamFile as ValidateCram</a></td><td>ValidateSamFile</td><td>Picard</td><td>Validates the CRAM file.</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="3-extract-nucleotide-flow-order"></a>3. Extract nucleotide flow order<a class="hash-link" href="#3-extract-nucleotide-flow-order" title="Direct link to heading">#</a></h3><p><strong>Sub-workflow name and link:</strong> <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/UltimaGenomics/UltimaGenomicsWholeGenomeCramOnly.wdl" target="_blank" rel="noopener noreferrer">UltimaGenomicsWholeGenomeCramOnly</a></p><p>The flow order is the order in which nucleotides are passed during sequencing. This information is captured in the header of the BAM and is extracted as a text file for downstream processing.</p><table><thead><tr><th>TASK name and WDL link</th><th>Tool</th><th>Software</th><th>Description</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.ExtractSampleNameFlowOrder</a></td><td>GetSampleName</td><td>GATK</td><td>Extracts the flow order from the BAM header into a text file that is used in downstream VCF post-processing.</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="4-calculate-quality-control-metrics"></a>4. Calculate quality control metrics<a class="hash-link" href="#4-calculate-quality-control-metrics" title="Direct link to heading">#</a></h3><p><strong>Sub-workflow name and link:</strong> <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/UltimaGenomics/UltimaGenomicsWholeGenomeCramOnly.wdl" target="_blank" rel="noopener noreferrer">UltimaGenomicsWholeGenomeCramOnly</a></p><ul><li><p><strong>Nested task name and WDL link:</strong></p><p><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineQC.wdl" target="_blank" rel="noopener noreferrer">UltimaGenomicsWholeGenomeGermlineQC.UltimaGenomicsWholeGenomeGermlineQC as CollectStatistics</a></p></li></ul><p>The workflow uses a contamination estimation that has been adapted to use only the highest quality SNP sites based on flow cycles and local realignment. This is different from the <a href="https://broadinstitute.github.io/warp/docs/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/README/" target="_blank" rel="noopener noreferrer">Whole Genome Germline Single Sample</a> pipeline in that HaplotypeCaller performs local realignment and feeds that output to VerifyBAM ID, which cleans up the alignment.</p><table><thead><tr><th>Subtask name and WDL link</th><th>Tool</th><th>Software</th><th>Description</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.HaplotypeCaller as HaplotypeCallerForContamination</a></td><td>HaplotypeCaller</td><td>GATK</td><td>Runs HaplotypeCaller using an interval list of variants with high allele frequencies (<code>contamination_sites_vcf</code>).</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.CheckContamination</a></td><td>VerifyBamID</td><td>VerifyBamID</td><td>Checks contamination in the HaplotypeCallerForContamination bamout file.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.CollectDuplicateMetrics</a></td><td>CollectDuplicateMetrics</td><td>Picard</td><td>Checks duplication metrics in the aggregated, duplicate-marked BAM file.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/Qc.wdl" target="_blank" rel="noopener noreferrer">QC.CollectQualityYieldMetrics</a></td><td>CollectQualityYieldMetrics</td><td>Picard</td><td>Calculates QC metrics on the duplicated-marked BAM.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.CollectWgsMetrics</a></td><td>CollectWgsMetrics</td><td>Picard</td><td>Collects WGS metrics on the duplicate-marked BAM using stringent thresholds.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.CollectRawWgsMetrics</a></td><td>CollectRawWgsMetrics</td><td>Picard</td><td>Collects the raw WGS metrics on the duplicated-marked BAM with commonly used QC metrics.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.CollectAggregationMetrics</a></td><td>CollectMultipleMetrics</td><td>Picard</td><td>Performs QC on the aligned, duplicated-marked BAM.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/Qc.wdl" target="_blank" rel="noopener noreferrer">QC.CheckPreValidation</a></td><td>custom script</td><td>python3</td><td>Checks chimerism and duplicate files for a given threshold using a custom python script.</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="5-variant-call-with-haplotypecaller"></a>5. Variant call with HaplotypeCaller<a class="hash-link" href="#5-variant-call-with-haplotypecaller" title="Direct link to heading">#</a></h3><p>The workflow implements initial variant calling with a version of HaplotypeCaller that handles flow-based data. This version replaces the classic Hidden Markov Model (HMM) with a flow-based likelihood model that more accurately accounts for sequencing errors present in the data.</p><table><thead><tr><th>Task name and WDL link</th><th>Tool</th><th>Software</th><th>Description</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/Utilities.wdl" target="_blank" rel="noopener noreferrer">Utilities.ScatterIntervalList</a></td><td>IntervalListTools</td><td>Picard</td><td>Splits the calling interval list into sub-intervals in order to perform variant calling on the sub-intervals.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.HaplotypeCaller</a></td><td>HaplotypeCaller</td><td>GATK</td><td>Performs initial variant calling on the aligned BAM file and outputs sub-interval GVCFs and a bamout file.</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="6-merge-vcfs-and-bams-and-convert-gvcf-to-vcf"></a>6. Merge VCFs and BAMs and convert GVCF to VCF<a class="hash-link" href="#6-merge-vcfs-and-bams-and-convert-gvcf-to-vcf" title="Direct link to heading">#</a></h3><p>The workflow performs multiple post-processing steps to prepare the VCF for downstream joint calling. The HaplotypeCaller GVCF outputs are merged into a single GVCF and then converted to VCF in preparation for this post-processing.</p><table><thead><tr><th>Task name and WDL link</th><th>Tool</th><th>Software</th><th>Description</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/GermlineVariantDiscovery.wdl" target="_blank" rel="noopener noreferrer">VariantDiscoverTasks.MergeVCFs</a></td><td>MergeVcfs</td><td>Picard</td><td>Merges the array of GVCFs from HaplotypeCaller into one VCF and index.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.MergeBams</a></td><td>MergeSamFiles</td><td>Picard</td><td>Merges the HaplotypeCaller bamout files into a single BAM file.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.ConvertGVCFtoVCF</a></td><td>GenotypeGVCFs</td><td>GATK</td><td>Converts to GVCF to VCF format in preparation for post-processing.</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="7-perform-vcf-post-processing"></a>7. Perform VCF post-processing<a class="hash-link" href="#7-perform-vcf-post-processing" title="Direct link to heading">#</a></h3><p>The workflow performs post-processing steps to prepare the VCF for downstream joint calling. First, it annotates the merged HaplotypeCaller output VCF with dbSNP variants, then it trains a model, a random forest classifier (<a href="https://www.biorxiv.org/content/10.1101/2022.05.29.493900v1" target="_blank" rel="noopener noreferrer">Almogy et al., 2022</a>), to distinguish between true positive variants and false positives. Next, the model is applied to filter variants in the VCF.</p><table><thead><tr><th>Task name and WDL link</th><th>Tool</th><th>Software</th><th>Description</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.AnnotateVCF</a></td><td>VariantAnnotator</td><td>GATK</td><td>Adds dbSNP annotations to the HaplotypeCaller output VCF and outputs a new VCF with index.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.AddIntervalAnnotationsToVCF</a></td><td>annotate</td><td>bcftools</td><td>Adds additional interval annotations to the VCF.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.TrainModel</a></td><td>Custom script (train_models_pipeline.py)</td><td>genomics.py3</td><td>Trains a model that can be applied for variant filtering to distinguish true and false-positive variants.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.AnnotateVCF_AF</a></td><td>view, index</td><td>vcfanno, bcftools</td><td>Adds gnomAD allele frequencies to the VCF for downstream metric calculation.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.FilterVCF</a></td><td>filter_variants_pipeline.py</td><td>genomics.py3</td><td>Applies a trained model to filter for high quality VCF variants.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/broad/UltimaGenomicsWholeGenomeGermlineTasks.wdl" target="_blank" rel="noopener noreferrer">Tasks.MoveAnnotationsToGvcf</a></td><td>annotate</td><td>bcftools</td><td>Move the annotations and the tree score from the filtered VCF to produces a final post-processed GVCF and index.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/germline/joint_genotyping/reblocking/ReblockGVCF.wdl" target="_blank" rel="noopener noreferrer">ReblockGVCF.ReblockGVCF</a></td><td>ReblockGVCF</td><td>GATK</td><td>Reblocks the annotated GVCF. For more information on why the WARP germline pipelines perform reblocking, see the <a href="https://broadinstitute.github.io/warp/blog/Nov21_ReblockedGVCF" target="_blank" rel="noopener noreferrer">blog on reblocking</a>.</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="8-outputs"></a>8. Outputs<a class="hash-link" href="#8-outputs" title="Direct link to heading">#</a></h3><p>Workflow outputs are described in the table below. </p><table><thead><tr><th>Output variable name</th><th>Description</th><th>Type</th></tr></thead><tbody><tr><td>output_gvcf</td><td>Final annotated, reblocked GVCF.</td><td>File</td></tr><tr><td>output_gvcf_index</td><td>Index file for the final annotated, reblocked GVCF.</td><td>File</td></tr><tr><td>output_vcf</td><td>Annotated VCF file.</td><td>File</td></tr><tr><td>output_vcf_index</td><td>Index file for the annotated VCF.</td><td>File</td></tr><tr><td>haplotype_bam</td><td>Optional merged, duplicated-marked BAM output.</td><td>File</td></tr><tr><td>haplotype_bam_index</td><td>Index for the optional merged, duplicate-marked BAM output.</td><td>File</td></tr><tr><td>output_cram</td><td>Final duplicate-marked CRAM file.</td><td>File</td></tr><tr><td>output_cram_index</td><td>Index file for the final CRAM.</td><td>File</td></tr><tr><td>output_cram_md5</td><td>MD5 sum for the final CRAM.</td><td>File</td></tr><tr><td>selfSM</td><td>Contamination estimate from VerifyBamID.</td><td>File</td></tr><tr><td>contamination</td><td>Estimated contamination from the CheckContamination task.</td><td>Float</td></tr><tr><td>filtered_vcf</td><td>Filtered VCF file.</td><td>File</td></tr><tr><td>filtered_vcf_index</td><td>Index for the filtered VCF.</td><td>File</td></tr><tr><td>quality_yield_metrics</td><td>The quality metrics calculated for the unmapped BAM files.</td><td>File</td></tr><tr><td>wgs_metrics</td><td>Metrics from the CollectWgsMetrics tool.</td><td>File</td></tr><tr><td>raw_wgs_metrics</td><td>Metrics from the CollectRawWgsMetrics tool.</td><td>File</td></tr><tr><td>duplicate_metrics</td><td>Duplicate read metrics from the MarkDuplicates tool.</td><td>File</td></tr><tr><td>agg_alignment_summary_metrics</td><td>Alignment summary metrics for the aggregated, duplicate-marked BAM.</td><td>File</td></tr><tr><td>agg_alignment_summary_pdf</td><td>Optional PDF of the alignment summary metrics for the aggregated BAM.</td><td>File</td></tr><tr><td>agg_gc_bias_detail_metrics</td><td>GC bias detail metrics for the aggregated, duplicate-marked BAM.</td><td>File</td></tr><tr><td>agg_gc_bias_pdf</td><td>PDF of GC bias for the aggregated, duplicate-marked BAM.</td><td>File</td></tr><tr><td>agg_gc_bias_summary_metrics</td><td>Bait bias summary metrics for the aggregated, duplicate-marked BAM.</td><td>File</td></tr><tr><td>agg_quality_distribution_pdf</td><td>PDF of the quality distribution for the aggregated, duplicate-marked BAM.</td><td>File</td></tr><tr><td>agg_quality_distribution_metrics</td><td>Quality distribution metrics for the aggregated, duplicate-marked BAM.</td><td>File</td></tr><tr><td>duplication_rate</td><td>Duplication rate.</td><td>Float</td></tr><tr><td>chimerism_rate</td><td>Chimerism rate.</td><td>Float</td></tr><tr><td>is_outlier_data</td><td>Boolean that indicates if the duplication rate and chimerism rate are above a specified threshold.</td><td>Boolean</td></tr><tr><td>sample_name</td><td>Sample name from aligned BAM header.</td><td>String</td></tr><tr><td>flow_order</td><td>Flow order (FO) from the aligned BAM header.</td><td>String</td></tr><tr><td>barcode</td><td>Barcode from the aligned BAM header.</td><td>String</td></tr><tr><td>id</td><td>ID from the aligned BAM header.</td><td>String</td></tr></tbody></table><h4><a aria-hidden="true" tabindex="-1" class="anchor anchor__h4 anchorWithStickyNavbar_31ik" id="using-outputs-for-downstream-joint-calling"></a>Using outputs for downstream joint calling<a class="hash-link" href="#using-outputs-for-downstream-joint-calling" title="Direct link to heading">#</a></h4><p>The outputs of the UG_WGS workflow are not yet compatible with the WARP <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/germline/joint_genotyping/UltimaGenomics/UltimaGenomicsJointGenotyping.wdl" target="_blank" rel="noopener noreferrer">Ultimate Genomics Joint Genotyping workflow</a>, but efforts to release the workflow are in-progress.  </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="versioning"></a>Versioning<a class="hash-link" href="#versioning" title="Direct link to heading">#</a></h2><p>All UG_WGS pipeline releases are documented in the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/broad/dna_seq/germline/single_sample/ugwgs/UltimaGenomicsWholeGenomeGermline.changelog.md" target="_blank" rel="noopener noreferrer">pipeline changelog</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="feedback"></a>Feedback<a class="hash-link" href="#feedback" title="Direct link to heading">#</a></h2><p>Please help us make our tools better by contacting <a href="mailto:warp-pipelines-help@broadinstitute.org" target="_blank" rel="noopener noreferrer">the WARP team</a> for pipeline-related suggestions or questions.</p></div><footer class="docusaurus-mt-lg"><div class="row"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Ultima_Genomics_Whole_Genome_Germline_Pipeline/README.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_">Last updated on <b><time datetime="2022-11-16T20:45:40.000Z">11/16/2022</time></b> by <b>kew24</b></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Smart-seq2_Single_Sample_Pipeline/Loom_schema"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« What&#x27;s in the Smart-seq2 Single Sample Pipeline Loom File?</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/README"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Whole Genome Germline Single Sample Overview Â»</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#introduction-to-the-ug_wgs-workflow" class="table-of-contents__link">Introduction to the UG_WGS workflow</a><ul><li><a href="#background-ultima-genomics-sequencing" class="table-of-contents__link">Background: Ultima Genomics sequencing</a></li><li><a href="#what-does-the-workflow-do" class="table-of-contents__link">What does the workflow do?</a></li></ul></li><li><a href="#set-up" class="table-of-contents__link">Set-up</a><ul><li><a href="#installation" class="table-of-contents__link">Installation</a></li><li><a href="#inputs" class="table-of-contents__link">Inputs</a></li></ul></li><li><a href="#ug_wgs-tasks-and-tools" class="table-of-contents__link">UG_WGS tasks and tools</a><ul><li><a href="#1-align-and-mark-duplicates" class="table-of-contents__link">1. Align and mark duplicates</a></li><li><a href="#2-convert-bam-to-cram-and-validate-the-cram" class="table-of-contents__link">2. Convert BAM to CRAM and validate the CRAM</a></li><li><a href="#3-extract-nucleotide-flow-order" class="table-of-contents__link">3. Extract nucleotide flow order</a></li><li><a href="#4-calculate-quality-control-metrics" class="table-of-contents__link">4. Calculate quality control metrics</a></li><li><a href="#5-variant-call-with-haplotypecaller" class="table-of-contents__link">5. Variant call with HaplotypeCaller</a></li><li><a href="#6-merge-vcfs-and-bams-and-convert-gvcf-to-vcf" class="table-of-contents__link">6. Merge VCFs and BAMs and convert GVCF to VCF</a></li><li><a href="#7-perform-vcf-post-processing" class="table-of-contents__link">7. Perform VCF post-processing</a></li><li><a href="#8-outputs" class="table-of-contents__link">8. Outputs</a></li></ul></li><li><a href="#versioning" class="table-of-contents__link">Versioning</a></li><li><a href="#feedback" class="table-of-contents__link">Feedback</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright Â© 2022 Copyright Â© Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.e360ee0b.js"></script>
<script src="/warp/assets/js/main.143bdc00.js"></script>
</body>
</html>